From 4eab3bc837ecea35dfff2dc6949136583ea816af Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Sun, 20 Dec 2015 15:12:11 +0200
Subject: [PATCH 0606/1240] tools: a3700: Initial addition of trusted boot TIM
 descriptors

- Revise the NTIM descriptor content
  (update date, remove unneded fields)
- Generate 16 CSK keys and 1 KAK key for future boot tests
- Add TIM and TIMN descriptors for trusted boot images
  making use of generated keys.
- Extend tim2img utility to support two input file names

Change-Id: I9f4cdff80e19f88a7ceda63f8834ca062f6fd7ba
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/26147
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 scripts/tim2img.pl               |  77 ++--
 tools/wtp/keys/CSK0.txt          |  55 +++
 tools/wtp/keys/CSK1.txt          |  55 +++
 tools/wtp/keys/CSK2.txt          |  55 +++
 tools/wtp/keys/CSK3.txt          |  55 +++
 tools/wtp/keys/CSK4.txt          |  55 +++
 tools/wtp/keys/CSK5.txt          |  55 +++
 tools/wtp/keys/CSK6.txt          |  55 +++
 tools/wtp/keys/CSK7.txt          |  55 +++
 tools/wtp/keys/CSK8.txt          |  55 +++
 tools/wtp/keys/CSK9.txt          |  55 +++
 tools/wtp/keys/CSKA.txt          |  55 +++
 tools/wtp/keys/CSKB.txt          |  55 +++
 tools/wtp/keys/CSKC.txt          |  55 +++
 tools/wtp/keys/CSKD.txt          |  55 +++
 tools/wtp/keys/CSKE.txt          |  55 +++
 tools/wtp/keys/CSKF.txt          |  55 +++
 tools/wtp/keys/KAK.txt           |  55 +++
 tools/wtp/keys/KeyGeneration.txt |   7 +
 tools/wtp/u-boot-ntim.txt        |  66 ++--
 tools/wtp/u-boot-tim.txt         | 836 +++++++++++++++++++++++++++++++++++++++
 tools/wtp/u-boot-timN.txt        | 126 ++++++
 22 files changed, 1980 insertions(+), 67 deletions(-)
 create mode 100644 tools/wtp/keys/CSK0.txt
 create mode 100644 tools/wtp/keys/CSK1.txt
 create mode 100644 tools/wtp/keys/CSK2.txt
 create mode 100644 tools/wtp/keys/CSK3.txt
 create mode 100644 tools/wtp/keys/CSK4.txt
 create mode 100644 tools/wtp/keys/CSK5.txt
 create mode 100644 tools/wtp/keys/CSK6.txt
 create mode 100644 tools/wtp/keys/CSK7.txt
 create mode 100644 tools/wtp/keys/CSK8.txt
 create mode 100644 tools/wtp/keys/CSK9.txt
 create mode 100644 tools/wtp/keys/CSKA.txt
 create mode 100644 tools/wtp/keys/CSKB.txt
 create mode 100644 tools/wtp/keys/CSKC.txt
 create mode 100644 tools/wtp/keys/CSKD.txt
 create mode 100644 tools/wtp/keys/CSKE.txt
 create mode 100644 tools/wtp/keys/CSKF.txt
 create mode 100644 tools/wtp/keys/KAK.txt
 create mode 100644 tools/wtp/keys/KeyGeneration.txt
 create mode 100644 tools/wtp/u-boot-tim.txt
 create mode 100644 tools/wtp/u-boot-timN.txt

diff --git a/scripts/tim2img.pl b/scripts/tim2img.pl
index dc11f56..2320cd1 100755
--- a/scripts/tim2img.pl
+++ b/scripts/tim2img.pl
@@ -1,18 +1,28 @@
 #!/usr/bin/perl
 
-sub tim2hex
+sub tim2img
 {
-	my ($in_file, $out_file) = @_;
+	my ($in_file, $out_file, $in2_file) = @_;
 	my $row;
 	my $index;
+	my $configs;
 	my $img_num = 0;
 	my @img_names = ();
 	my @img_offset = ();
+	my @cfg = ();
+	my $cfg_idx;
 
-	unless(open ($cfg, "<$in_file")) {
+	unless(open ($cfg[0], "<$in_file")) {
 		print ("Failed to open input file $in_file\n");
 		goto ERROR;
 	}
+	$configs = 1;
+
+	unless(open ($cfg[1], "<$in2_file")) {
+		print ("No input file for TIMN is supplied\n");
+	} else {
+		$configs++;
+	}
 
 	unless(open $dst, ">$out_file") {
 		print ("Failed to open output file $out_file\n");
@@ -24,42 +34,43 @@ sub tim2hex
 		goto ERROR;
 	}
 
-	# Parse configuration file
+	# Parse configuration file(s)
 	# Find a number of boot images and their names
-	while ($row = <$cfg>) {
-		chomp $row;
-		($name, $value) = split(":", $row);
-		# trim spaces from both ends
-		$name =~ s/^\s+|\s+$//g;
-		$value =~ s/^\s+|\s+$//g;
-
-		if ($img_num == 0) {
+	$index = 0;
+	for ($cfg_idx = 0; $cfg_idx < $configs; $cfg_idx++) {
+
+		while (defined($row = readline($cfg[$cfg_idx]))) {
+			chomp $row;
+			($name, $value) = split(":", $row);
+			# trim spaces from both ends
+			$name =~ s/^\s+|\s+$//g;
+			$value =~ s/^\s+|\s+$//g;
+
 			if ($name eq "Number of Images") {
-				$img_num = $value;
-				$index = 0;
-				print ("Total number of images to process - $img_num\n");
-			}
-		} else {
-			if ($name eq "Image Filename") {
-				$img_names[$index] = $value;
-				print ("is $img_names[$index]\n");
-				$index++;
+					$img_num = $value;
+					print ("Total number of images to process in file[$cfg_idx] - $img_num\n");
+			} elsif ($name eq "Image Filename") {
+					$img_names[$index] = $value;
+					print ("is $img_names[$index]\n");
+					$index++;
 			} elsif ($name eq "Flash Entry Address") {
-				$img_offset[$index] = hex($value);
-				$offs_str = sprintf("%08x", $img_offset[$index]);
-				print ("$index Image at offset $offs_str ");
+					$img_offset[$index] = hex($value);
+					$offs_str = sprintf("%08x", $img_offset[$index]);
+					print ("$index Image at offset $offs_str ");
 			}
 		}
-	}
 
-	close($cfg);
+		close($cfg[$cfg_idx]);
+	}
 
 	if ($img_num < 1) {
-		print ("The \"Number of Images\" field was not found in $in_file\n");
+		print ("The \"Number of Images\" field was not found in input files\n");
 		goto ERROR;
 	}
 
 	#--------------------------------------------------------------------------
+	# Create output from collected images data
+	#--------------------------------------------------------------------------
 
 	for ($index = 0; $index < $img_num; $index++) {
 
@@ -106,12 +117,13 @@ ERROR:
 
 sub usage
 {
-	print "\nConvert TIM/NTIM binary files to flash IMG format\n";
+	print "\nConvert TIM/NTIM and TIMN binary files to flash IMG format\n";
 	print "\nAccording to TIM/NTIM configuration file\n";
-	print "\nExample  : tim2img.pl -i tim.txt -o Image.img\n";
+	print "\nExample  : tim2img.pl -i tim.txt -n timn.txt -o Image.img\n";
 	print "\n";
 	print "Options:\n";
 	print "\t-i\tInput TIM/NTIM configuration file in text format\n";
+	print "\t-n\tInput TIMN configuration file in text format\n";
 	print "\t-o\tOutput file in flash IMG format\n";
 	print "\n";
 }
@@ -121,9 +133,9 @@ use strict;
 use warnings;
 use Getopt::Std;
 
-use vars qw($opt_o $opt_h $opt_i $opt_s);
+use vars qw($opt_o $opt_h $opt_i $opt_s $opt_n);
 
-getopt('o:i:s:h');
+getopt('o:i:s:h:n');
 
 if ($opt_h)
 {
@@ -148,10 +160,9 @@ unless ($opt_o) {
 }
 
 
-if(tim2hex($opt_i, $opt_o)) {
+if(tim2img($opt_i, $opt_o, $opt_n)) {
 	printf "\nError: Failed converting image\n\n";
 	exit 1;
 }
 
 exit 0;
-
diff --git a/tools/wtp/keys/CSK0.txt b/tools/wtp/keys/CSK0.txt
new file mode 100644
index 0000000..1ac0273
--- /dev/null
+++ b/tools/wtp/keys/CSK0.txt
@@ -0,0 +1,55 @@
+Key ID: CSK0
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x84724B4D 0xF58CA3E8 0xD379F131 0x1ABFF997
+0xC8D58E4C 0x5CA18F30 0xB739BA64 0x5938611F
+0x8EE5D2B6 0x0EF040E8 0x153DB054 0x1A22F8BE
+0xCBB9BCEF 0xC2793935 0xF133342F 0xA31B788C
+0x44AEA9C7 0x0F255B5E 0xA8D0DBDB 0xC2CD9DC3
+0x34DCC5A1 0xF5812DEF 0x58C2D51C 0x879562B1
+0x1F2A5A8B 0x4884300D 0x51F9A7DF 0xB8229036
+0xC73EA8B4 0xDDA0AF27 0x9CE80B9F 0x393B0C79
+0x5630EA68 0x558D53AE 0xCCF630EF 0x72247090
+0xBCB40C87 0x3DC04183 0x9FB8E19F 0x7BD96878
+0x534168EA 0x20FA2569 0x135829A3 0x0A34F9BD
+0x5994A70C 0x0D8E43E0 0xE352D564 0x411D3AA5
+0x4F666B2A 0x5CCFFADA 0x0B14C219 0x82E9B28C
+0x1AFDC5A8 0x2E699FCC 0x1D79E540 0x457CE2EB
+0x4D56DACE 0xAA900176 0x366DAFDC 0x1F895984
+0x4EBC5283 0x5668D9E4 0x889D174B 0x8BB67F04
+RSA Private Key:
+0xA95F8393 0x87262FD6 0x6F520684 0x97F6C770
+0x7A5B7B65 0x89D4E137 0x20E462C8 0xB2236325
+0xBC342B78 0x8BA72C74 0x40E14F5A 0xDEF950A3
+0x396451F0 0x93EBD6E5 0xBFCB0F02 0xC93D6B1A
+0xC0B55BCE 0x3CBB0781 0x1637A0AC 0x182D0352
+0xA33510CE 0x6A3CEFFF 0x596975F8 0xA37E67A3
+0x9BAC0729 0x020A8450 0xDB3FB9D4 0xDF2F0BA6
+0xF80F91D6 0xF00506FE 0x20DC15DA 0x7DC8537F
+0xE4209C44 0xE3B38D1E 0xDDF975F4 0xF6C2F5B5
+0xD322B304 0x292AD657 0xBFD0966A 0xA7E645A5
+0x8CD645F1 0xC0A6C39B 0x623AC66C 0x06CDFBD3
+0xE6631A08 0x5E5ED7EA 0x423738ED 0x80BE2719
+0x8A444771 0x3DDFFC91 0x5CB88166 0xAC9BCC5D
+0x11FE83C5 0x74466A88 0x13A698D5 0x2E5341F2
+0xDE39E734 0x71B5564E 0x24491FE8 0xBFB0E658
+0x89D2E1AC 0x399B3BED 0xB068BA32 0x5D2454AD
diff --git a/tools/wtp/keys/CSK1.txt b/tools/wtp/keys/CSK1.txt
new file mode 100644
index 0000000..6098ca9
--- /dev/null
+++ b/tools/wtp/keys/CSK1.txt
@@ -0,0 +1,55 @@
+Key ID: CSK1
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x80EED371 0x0AB09B9B 0xA8BE98CD 0xB3246236
+0x4518BCB8 0x497B8AE2 0xB7B5180C 0x71C9A948
+0x458692E5 0x855AD78F 0x0F118A08 0xCBB5B7F0
+0xB4A915FD 0xD213B4A4 0x6526A4AB 0x7A5765D1
+0x147FA13D 0x544031DB 0xB9F08C82 0x2AA3B954
+0x318DA699 0x2547A92C 0xEBB021A4 0x41B46DFB
+0x420261F2 0x53C2E1E0 0xB396E865 0x21745D50
+0xC85A7C8D 0x9B1953C0 0x06B64867 0xD2033737
+0xAB830B2C 0xF091685F 0x8A8C7F6E 0x7AE0436E
+0xC3FAFD35 0x34A1BC5B 0xBCCE1684 0x2A5B3168
+0x378DBDF4 0x20B5DE97 0x2E11AD52 0xF43E6531
+0x6608353F 0x4D11B545 0x7E9AEA03 0xD1299F13
+0x995C3700 0xEBF5CEDF 0xABDC0164 0x50B8250B
+0x387E5E50 0x62258E08 0x019F98D3 0xC530F7AB
+0x9B120CAF 0x52335B28 0x3B8896A0 0xEB5C6D17
+0xE67F80DB 0xD5A4B203 0x043872B6 0x9BD47764
+RSA Private Key:
+0xB5823489 0xAF89A32B 0x5072350D 0x21875BB8
+0x18637EDE 0xD8A962C4 0xFF2D58DF 0xDA55D37F
+0xCE850F34 0x275AD017 0x4951BA24 0x0D4756FE
+0x4A8A5053 0xD704A1B2 0x2E5D6E3E 0xE046FD51
+0x9A35A0BD 0x05619706 0x55BB76C8 0x19E4444C
+0x1AA54409 0xFAE7C02B 0x84BD2862 0xBCD24485
+0x2614411B 0x1D255FDF 0x73040962 0x3B73B6FF
+0x3948CB15 0x44947E9A 0x296CA9D6 0xBDC8D470
+0x66E839E6 0x29F0D839 0xECBAB2DC 0x16868EDB
+0xA8C9CB20 0x1F943DD0 0xA47BA71C 0xB303840B
+0x54883EC5 0xAD39EBF4 0xE8710197 0xC5BF0983
+0xD6D1B98C 0x6171065C 0x4BF68C68 0xB0B292A5
+0x8F375433 0x5A6048EC 0x00B73409 0x96D4E307
+0xEEB23896 0x07B02204 0x00F95BB2 0xA983C7CD
+0x903E079C 0xFE1ED04B 0xF085272C 0xF39DDB0D
+0x8A4C8083 0xB362D135 0x9C21DE6D 0x5D7F7AD5
diff --git a/tools/wtp/keys/CSK2.txt b/tools/wtp/keys/CSK2.txt
new file mode 100644
index 0000000..ddcf443
--- /dev/null
+++ b/tools/wtp/keys/CSK2.txt
@@ -0,0 +1,55 @@
+Key ID: CSK2
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x0000000B 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0xA909FF45 0xA432D24A 0x1E9A2FE9 0x062A3CC5
+0xC5286B43 0x5F599935 0x5342A93F 0x53BE93E6
+0x978DBC01 0x29F90652 0xFEB50E31 0xA40EFB45
+0x1FD4AC41 0x71C209AD 0x7DA7868D 0x224A835F
+0x4E8B3F95 0xD4D60A04 0x58CFB7F2 0x8861C7D3
+0x87704D6B 0x4824ABDB 0x92020A1D 0x8F08D3DF
+0xC1A3A6F5 0xE200120C 0xBFEE8322 0xB79FF73E
+0x8DEFFF8E 0x3B49DC35 0x4B7530BA 0xBE1DD02E
+0x76743693 0x11FE0B29 0x98E82CD5 0x324A8FC6
+0xDA1685C8 0x4E1F9224 0x75488EAB 0x35B0A53F
+0xEDAD6DB0 0xB9ECF27E 0xB8ACDB84 0xBEEA13D5
+0x1C160A24 0x6AC615F3 0xEC5BBC36 0x09F04D5E
+0x4BEA748E 0x75A8BA08 0xA1C096CF 0x301FFB75
+0xE2DE156F 0xFA694BE8 0x41EC3BC2 0xA52AEB45
+0x0271D5B2 0xFBBF692B 0x4236E95D 0x0E105EA1
+0x82E6A06C 0xEDE2FA67 0xAB731C64 0xD380DE78
+RSA Private Key:
+0x279E3863 0xB4AD1423 0xF7F45227 0x075B59E7
+0xC47E7EC1 0x79CB3E94 0x91C7D9E5 0x3CC7B7CE
+0xA29E39C2 0x850B152B 0xAB18E169 0x904F4D79
+0xCFA79185 0xC86E1168 0x3D1E1299 0xD0B81616
+0xBBA5DC34 0x29784235 0x76D4B3DE 0xA933A732
+0xBC4950E7 0x14F797D9 0x8979CD3A 0xC645F577
+0xEBA5F517 0xB4A53E1A 0x54A9E536 0x3DABF3DA
+0x585325E6 0x68E00D46 0x31BAE246 0xD6651EA1
+0xC8C6BD3E 0xF9157E82 0x16A485D8 0xA2153CE3
+0xC6430541 0x0133F938 0xB07081B3 0x76A09639
+0xA98663B7 0x061D398A 0x04FA3BED 0xDC1A9DAB
+0xD3B6F1F2 0x8F9CCE22 0x1CB07C8E 0xDA7D5D9C
+0xFF323B69 0xDF5391D8 0xF02371D3 0x147A2A6A
+0x140FB665 0xFAEB5C48 0x99054D99 0xDBF878CA
+0xBC677C73 0xCD96BCB2 0x82035FC9 0x0CC91035
+0xA58BD7A8 0xEF88868C 0x84973115 0xC0469BB3
diff --git a/tools/wtp/keys/CSK3.txt b/tools/wtp/keys/CSK3.txt
new file mode 100644
index 0000000..f83f680
--- /dev/null
+++ b/tools/wtp/keys/CSK3.txt
@@ -0,0 +1,55 @@
+Key ID: CSK3
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0xCC14894B 0x1D54EFAB 0x2C006ECC 0xECBA39B5
+0x386B9504 0xE5127FA9 0xBBAF3A06 0xB351C24B
+0x90A1B218 0xDB877D1E 0xFAAECD8C 0x2ABDC223
+0x88AB687E 0x3BB63A7A 0xE2CECAA9 0x9F51C0D8
+0xF55351ED 0x68C4C09E 0xF4FED718 0x111E24B9
+0x8D938BEB 0x8ECF27F9 0xBBE7D8D5 0x1AC4F691
+0xD54FBDB1 0x18A6D6BE 0x9619F30B 0x96461657
+0xDF2961EC 0x3AE58790 0x88ECC5BD 0x181A4B15
+0xED03EB8E 0xB365ED3E 0x7C95DCEA 0xE13FEF7A
+0x046073C2 0xDF910B26 0x3F629BFD 0xDA8AD3F3
+0xE3FBCFD8 0x49338817 0x8BBF17BD 0x02AAEAFA
+0x4678A53E 0xB7D51770 0xBF853DCA 0x75487D2D
+0xD57ECF69 0x9E444CF0 0x0B7B1D0D 0x38C1DEE9
+0xEDC2BDB1 0xDF03CEA1 0xDDD31093 0x0317A73D
+0x02EB8F74 0x5519D9F9 0xA3501FDE 0xC4BA749D
+0xA82D50A6 0x0E42C2C6 0x0E56FC52 0xA533A925
+RSA Private Key:
+0xC943324B 0xCE30A122 0x4A1B377D 0xA80E4F16
+0xFFB441C8 0x196E7696 0x32ADDC6C 0xF2A97A06
+0x754B38D7 0x0339DA63 0xA10C8866 0x6A96C338
+0x883DA2BA 0x12D95E0A 0xA8C66443 0xF31C28AC
+0x02217A27 0x44A5611A 0xD47EEDC5 0xBF607912
+0x45DCE74B 0x369C0D05 0x125C8DBE 0x520AA3AE
+0xA2D82F88 0x96B5F30E 0x14812111 0x5F349720
+0xC636D203 0xE68D9D63 0x1E7FCC03 0x5086A930
+0x9E029D08 0x779948D4 0xA863E89C 0x40D54A51
+0x02EAF7D7 0x950B5CC4 0xD4EC67FE 0x3C5C8D4C
+0x97FD353B 0x86225ABA 0x5D2A0FD3 0xAC71F1FC
+0xD9A5C37E 0x2538BA4A 0xD5037E87 0x4E305373
+0x8E548A46 0x142D88A0 0x5CFCBE09 0x25D6949B
+0x492C7E76 0xEA0289C1 0x3E8CB5B7 0x57651A29
+0xAC9D0A4D 0xE3669150 0x6CE0153E 0xD87C4DBE
+0xC5738B19 0xB42C81D9 0x5EE4A836 0x6E2270C3
diff --git a/tools/wtp/keys/CSK4.txt b/tools/wtp/keys/CSK4.txt
new file mode 100644
index 0000000..d17810c
--- /dev/null
+++ b/tools/wtp/keys/CSK4.txt
@@ -0,0 +1,55 @@
+Key ID: CSK4
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x1CE759B9 0x4AFF4832 0x9CFA2713 0xCD988395
+0xCE44E7B2 0x1BEDD8C1 0xFBB53605 0xA042C260
+0x9BE6075A 0x02E3372E 0x3E1DAE9C 0xAB6C0387
+0xFE6C24F8 0x7335FEE3 0xB43A2AC4 0xE7BB7C7D
+0x388349C2 0xFD5671CE 0xA8364768 0x5DC26EE9
+0x0B3DFDAE 0xD803592A 0x26AB97BF 0xE3EB4EBB
+0x4CB562E4 0xAB5FB74E 0x3F200C72 0xE569DF3E
+0xB310E40C 0xA7E87BCD 0xEE99B2F1 0xAE69FFF9
+0xD47A88E7 0xB0EBEB19 0x21D613DA 0xF84E7A42
+0x308C1F39 0x0A3C4E8D 0x83977B36 0xBE02DA3E
+0xE36AF13A 0x098C57C5 0x5867DE7B 0x64FA0437
+0xA5CD852B 0xEC154C71 0x5073B641 0x231BFCD4
+0x05EEDE28 0xF64E1471 0x6EA07407 0x7D0430ED
+0xD900FC06 0x288FC558 0xB2A95E0D 0x451A2DE7
+0x5E51F306 0x3F3E6FA2 0x7338E97A 0xC4BAA7BB
+0x0C4A7ED5 0x41842DCF 0x03E84F51 0x981B690A
+RSA Private Key:
+0xE31D2CCD 0x1316E895 0x3120EBB9 0xD1428696
+0xDB92FB1A 0xE66C5BF9 0xFBC1D528 0x84F9E610
+0xA3DF24F6 0x2B01C52F 0xADE10656 0x9FB750CD
+0x699A123D 0x7CBA245A 0x9193483C 0xE881FDC7
+0x9EA7A7A0 0xE07D4CD6 0x260DE221 0xC084FB93
+0xF3BF1089 0x316D3147 0x0E7935FC 0xB7A5BC14
+0x7F6E45EB 0x05A11DB2 0x3D8E9D12 0x2A2A5BB8
+0xB0A17CD6 0x54B2529E 0x781A2640 0x487F2AA5
+0xDD2ED3EB 0xF3EFEF47 0x4E44DCAE 0xC6A52E9B
+0x5A09B294 0x08303ED7 0x3612C8F8 0xCB357B65
+0x4F88C0FB 0x3AD6AC9E 0xAD1FE52F 0x83FB3692
+0xEB0AD0EF 0x5677705A 0x405C91CE 0x4F499710
+0x37F24B53 0x5EA4DD27 0x8BB39006 0x64035A57
+0xAD9A6338 0xBA0C9DE0 0xF5544B3D 0xD0E1BE52
+0xB1DB28D1 0x3298594E 0x8F60BAC8 0x03C8862F
+0x703B98AB 0x679CF172 0x365372A7 0x79AF873B
diff --git a/tools/wtp/keys/CSK5.txt b/tools/wtp/keys/CSK5.txt
new file mode 100644
index 0000000..1f85966
--- /dev/null
+++ b/tools/wtp/keys/CSK5.txt
@@ -0,0 +1,55 @@
+Key ID: CSK5
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x0000000B 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0xB42F42BD 0x2D6BB77C 0x3DBBF682 0x97E79D79
+0x112F41CD 0x1440BA6E 0x72B8CBA1 0x9F3AC71C
+0xAFFB5B83 0xCF6CE01A 0xA451F1C4 0x840F8B39
+0x47F5B6B4 0x52D24D6B 0xF940FCE5 0x938714CD
+0xAE8ED49A 0x011C7F4D 0x20A0EAB3 0x854433C6
+0xFD25B036 0x5F402E71 0x542B5679 0xCACC93CB
+0x1DDB8ADC 0x426B1636 0x99944330 0xF8F9CEDF
+0x013B1D84 0x9AB89FF3 0x025DC47D 0xF092EF75
+0x8E61250C 0x6C07157B 0x60F43C83 0x29936A4E
+0x6BF5A125 0x5B55EAEC 0x15828E0C 0x48E23DF3
+0xFB41A5E8 0xC48C9CB4 0x217FDBEC 0x837D569B
+0x859FEB98 0x249012F9 0xCC50C508 0xA6253F45
+0x236BC1A8 0xA7B3062F 0x944A9B4E 0x82D973E2
+0x7C0BC74E 0x38BCAEE8 0xF59EBEE0 0x64376CD6
+0xB253A4D0 0xCFCF01A8 0x073FB688 0x27B3ED9B
+0x94C016CC 0x0783D671 0x6FD68A39 0x8869DD7C
+RSA Private Key:
+0x42B0C39B 0x20F8CE85 0x1D5241EE 0x0596EDC9
+0xA342BCF6 0xABF5CEE8 0x184EDF12 0x1F173283
+0xF1B326D8 0x14F0803B 0xAF45B0B6 0xC5BF237D
+0x6EAD6A49 0xF768D80E 0xC5F15E53 0x0D316F8F
+0x2FE4219D 0x2C5911C4 0x351975C4 0x388BC228
+0xD314F0E2 0xE5FF9928 0xA30B63A2 0xCB9FCBD2
+0x683A925D 0x5F9D5C9C 0x0D0C6962 0x52C34301
+0xC06776C0 0x8E916C3C 0xF6827958 0xCDE319F6
+0xF0921434 0x0C60F471 0xD7CB09BC 0xA2506887
+0x5228E388 0x31D1C5F5 0xF475C1D8 0x564CDBF8
+0x890C890A 0x99C10FA8 0xFAFFEC52 0x76445DC8
+0xA5FA236A 0x13F17EB6 0xFB14C890 0xB7B739C8
+0x9EF4F544 0x15A777BC 0x399D0EE5 0x30198507
+0xE6920F9F 0x4D7E30DB 0x576DDC7A 0x1F640CD2
+0xECE7CE43 0x14425DFE 0xD568921C 0x2CEDC76B
+0x39DD23B5 0xBE47E955 0xDFE962AA 0x4A684A43
diff --git a/tools/wtp/keys/CSK6.txt b/tools/wtp/keys/CSK6.txt
new file mode 100644
index 0000000..1f93959
--- /dev/null
+++ b/tools/wtp/keys/CSK6.txt
@@ -0,0 +1,55 @@
+Key ID: CSK6
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x3F236983 0xEBDB683B 0xDDA3648F 0x914B5800
+0x31B22E27 0x174F42D6 0x9FCF6B74 0x702CECF7
+0x64DF7FBA 0x402FDB7F 0x9F508CBD 0xA297B410
+0x0BDA7224 0x8531ADDE 0x5B3B720A 0x888138ED
+0xD12A0E97 0xFDA8AF45 0x4EC9073D 0xEBB56CEF
+0xFD1CB219 0xDE792A64 0x8D2B323B 0x1853D025
+0x789D2E0E 0x14546422 0x856E4789 0xF4858605
+0x3EC6F74A 0x8E6E0FDA 0x591A6D47 0xAD05DC1A
+0xA50D3F8D 0x8B98A196 0x08434948 0x767F9DE6
+0x2D7A6D73 0xF7E5EED0 0x53A46A03 0x008457A2
+0x4F086954 0xC98D0C87 0x37C60DD9 0x5A3AF863
+0x8B535028 0xAF3FC78E 0x0AC0CB91 0x63BD1E08
+0x0E9013FC 0xC86D4C3C 0xE0B434F5 0x448790F3
+0x535EE091 0xD2C23186 0xEDAF39A9 0x16AF5681
+0x46DB7772 0x5F7EC0E6 0x0A96AF66 0x432713A5
+0xD813F959 0xB874402E 0x3A4A2145 0x86826C46
+RSA Private Key:
+0x9661DE2B 0x528B821D 0x641062DD 0x05CAB8E4
+0x7BECD992 0x1FE10DCA 0x96D3AE7A 0x169B59F5
+0x53A42496 0xC9693EEF 0xEF8EAB95 0x7810798F
+0xA9BF190C 0x45B98E09 0xF0ECE0B5 0x8E33F103
+0xB7F2F23C 0x5B7671AA 0xE78606C8 0x182DABBF
+0x102A5773 0x83575AA4 0xA5AB8701 0xB53E380D
+0x855DB7AE 0xBCC5A7B9 0x9506953F 0x5ED68C99
+0xB1B24D9D 0x01AA4684 0xCE2BC96F 0x25085203
+0x6E08D508 0x07BB1664 0x058230DB 0xF9AA6944
+0x1E519E4C 0xFA9949E0 0x37C2F157 0x00583A6C
+0x34B04638 0x865E085A 0x252EB3E6 0x3C275042
+0x5CE23570 0x74D52FB4 0x5C8087B6 0x427E1405
+0x09B562A8 0xDAF38828 0x95CD78A3 0xD85A60A2
+0xE23F4060 0xE1D6CBAE 0x9E74D11B 0x6474E456
+0x84924FA1 0xEA548099 0x5C6474EE 0x2CC4B7C3
+0x900D50E6 0x25A2D574 0xD186C0D9 0x59AC482E
diff --git a/tools/wtp/keys/CSK7.txt b/tools/wtp/keys/CSK7.txt
new file mode 100644
index 0000000..c577bc9
--- /dev/null
+++ b/tools/wtp/keys/CSK7.txt
@@ -0,0 +1,55 @@
+Key ID: CSK7
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x8FA6AA23 0xCCF84CA4 0xFEB69002 0x996E098C
+0xB909AE3D 0x2D9DF389 0x7FD369A3 0x7FF61B61
+0x4D8E4730 0xEC43162D 0x2184D5F8 0xA05A0529
+0xAECB5125 0x4E59A548 0xE9247D4C 0x0053218C
+0x2F5084E0 0x8CD04BA5 0xACE14EE3 0xA24BA650
+0x9492DE2F 0xE76E7164 0x58D84806 0xAB0DEC2E
+0xAF39B139 0xB585F725 0xFF9B8A42 0x46F3F169
+0xE22570CE 0x6681864E 0xF2713D5E 0x7FFFB5AC
+0x8C2E22F7 0xB8BECCF8 0xB80AC29D 0xD8A61BD5
+0x4B51BC1B 0xF5895F18 0x6E0C64C8 0x356CADDD
+0x973136D7 0x0B523327 0x9252B3C7 0x537E92A1
+0x79FC3EC6 0xD7A627D1 0xF6999E76 0xB8FEB250
+0x40DD0A0E 0xF70AD291 0x6D6CE402 0xD3EDC239
+0xA6BCB44D 0xDFDD1C0C 0x0FC00C2F 0x015FD8A4
+0x441669BB 0xD63FA75F 0x6106EA2E 0xC5C433BC
+0x34DE89D1 0xD7360B9E 0xCD3CC15A 0xBD689D5F
+RSA Private Key:
+0x7363F005 0xFC5952F0 0x7FC37C3F 0x4A73D2FB
+0xE30D1920 0x2B5FB269 0xC99F98C2 0x4996560C
+0xA96A58D1 0x4C813B63 0x9761DFD8 0x0EECDC8C
+0x440CF47F 0xED693639 0x3E78C8EE 0x8924B0B8
+0xBDAE26D7 0x7C6EE8D0 0xFE4ADF46 0x5074A194
+0x8678B798 0xB5BF217D 0xF42AEC83 0xF01BC773
+0x5889B28F 0xA18A3A2A 0x0DDA364C 0x104C13B7
+0xFBA3816A 0xF9D3AA52 0x802736E4 0x7653C9C8
+0x541BAE93 0xD53F47C8 0xD4D34191 0xB5307719
+0xC6CAA410 0x93526C41 0x0ED43C78 0xECDACEB8
+0xC11D874D 0xA0648517 0x57CB38AA 0x987F24C7
+0xE2CA8C10 0xB496E4B0 0xC728F8AD 0x0898D163
+0x5A1E393C 0x9439B18A 0x0E7488CE 0xE58EA7BC
+0x30D79F61 0xECB7DDA1 0xD640074F 0x34064EC8
+0x28DA3F70 0x1A263139 0x0704261C 0x10428571
+0xB95252B1 0x4DED3A2B 0x7B247403 0x71A52B39
diff --git a/tools/wtp/keys/CSK8.txt b/tools/wtp/keys/CSK8.txt
new file mode 100644
index 0000000..ab98493
--- /dev/null
+++ b/tools/wtp/keys/CSK8.txt
@@ -0,0 +1,55 @@
+Key ID: CSK8
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x6EB96073 0xD83975B3 0x6BB3FA48 0x77F435AB
+0x1CB0C1DC 0x7B9C076D 0x704A7D7B 0x1DBF8D59
+0x200DB680 0x420439A8 0x33F7BBBC 0x2CB397DB
+0x08FA9EBB 0x2EA446DB 0x46AAA79D 0x5B527159
+0xFA9CC944 0x8CE39B34 0x6E80DF9F 0x4DCA4310
+0x3CC2DD75 0x262A2F72 0xAB74CB32 0xE1153DD3
+0xEE966E87 0x09DBD071 0x0156F389 0x7EDB7DD4
+0xFBA01311 0xD1805968 0xCBE3BFF7 0x7ADFAD6E
+0x77CBCCEA 0x7C89F4F8 0x82259AFA 0xEE472314
+0xEFB72065 0xC32E1049 0x43EEE9F6 0x372E5004
+0x4613B37C 0x3B09569A 0x5588FBDE 0x93979FEA
+0x70485A1D 0x2BC862FC 0x0A970E72 0x33409DE5
+0x4B6C97AC 0x88204426 0x2EDB395F 0x9C689899
+0x64126A7F 0xAC9A4CF7 0x126A2041 0x404AB206
+0x64B85FD9 0xA7BD445A 0x12CA7C28 0x4955918E
+0x40B8BF7D 0x45CB59F2 0xE5A8A467 0x9F94C876
+RSA Private Key:
+0xF2E4104D 0x5D35CBC5 0x608A724B 0xA41DF140
+0x8898131C 0x3FBA5623 0x3564DBA2 0x4EDF1568
+0xCCAFD738 0xDE7490EA 0x70CEEE15 0x53E15FCB
+0x3851B3CB 0xF88E36AB 0x485FBD8D 0x7C569E6F
+0x297D1E20 0x8879E78A 0xE5FB0ED2 0x6FED01AC
+0x9E28F2E7 0x6DF508AB 0xA88BF5DA 0x3B1C6A4B
+0xF0FD891C 0x3D1C201C 0x89A88E92 0x5EBFEB2F
+0x96A70EF6 0x9282BE52 0xDA48B7C4 0x833E8952
+0xC63CA3ED 0x63A190C6 0x9B5148C8 0xBE9F4F43
+0xBFC5B384 0x9C24D9D4 0x032587F8 0xF8F1D99D
+0x6B42F5FC 0xFC07787B 0xDE072FE4 0x42DFB321
+0xC039E1B1 0x2306B596 0xD545A528 0x29007E50
+0x6F8A12F0 0x6CE69CEB 0x8BE2944C 0xE386E07A
+0xE9A85532 0xF07B70C5 0xA854E69A 0x336EF4D1
+0x83C6B314 0x8631037B 0xDBD53020 0x0777A7A4
+0xCD609931 0x6B0914C1 0xB7BA1D1F 0x7FAA39F8
diff --git a/tools/wtp/keys/CSK9.txt b/tools/wtp/keys/CSK9.txt
new file mode 100644
index 0000000..5585a2e
--- /dev/null
+++ b/tools/wtp/keys/CSK9.txt
@@ -0,0 +1,55 @@
+Key ID: CSK9
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x1A391637 0xF3981418 0x50F8BA85 0xE6765B83
+0x8E69891F 0x8DB23893 0xEE0FCA94 0x7ED0D5B0
+0xA491F91A 0x9B5E8196 0xD9344934 0xEBE12AD5
+0x81D1C27E 0xB20BA288 0x04963930 0x0E7A293B
+0xE7D56DCB 0x8F65E162 0xF6BBDEE0 0x86AEA562
+0x3D08C23C 0x768EA975 0x377D71CC 0x17B9BC0B
+0x6AB327C3 0x407B7862 0x24FF1095 0xFDB888D9
+0xC422FB8B 0xB87FDE7C 0x05D83774 0x1AA1ACB6
+0xAAA7B491 0xF054D003 0xAB637711 0x1FD7D1BE
+0xE9966138 0x077024D3 0xE4E8BBE8 0x2E962E07
+0xECF5A346 0xF39676A2 0xE702A229 0xDBF8B6F5
+0xDDDD3D6F 0x7A5154D7 0xE040D453 0xCC0F85C9
+0x34FA581F 0x95D4B82D 0x6A797D75 0xE1553B71
+0xD47EA44C 0xCE7D12DE 0x734843F3 0x05F18F09
+0xC4CC8D7E 0xD4D0B8A8 0xE48867A4 0x061A328A
+0x939A6DF2 0x8D715E49 0x8F89220D 0xE8B41609
+RSA Private Key:
+0x4070972D 0xD247ED99 0x48438CD3 0x9FCD97E9
+0x65905316 0xF88F802F 0xFA5CCD2D 0xDFD0586D
+0x7DC75854 0xDB5D3F73 0xABF8082A 0x6CDF4927
+0x29773EE3 0x87711633 0x62383E91 0xBE79DE31
+0x4FAF8089 0x73DA5205 0xB27F67AC 0x582731DB
+0x98637567 0x5F51F7D7 0x97EC6F7A 0x63EE95B1
+0xC7A60308 0x71053733 0x68059479 0xFBF5ACB7
+0xC769C007 0x6703D4A4 0x3F7C7F7A 0x5B922800
+0xBBB95D3F 0xF3770CCF 0xEF82C5A7 0x7FDFDAFE
+0x54784DC6 0xD2C01D76 0x50BA2FEC 0x8BAB5806
+0xF0C4829E 0x8FAB921B 0xB8CEE821 0x7CC6F8C4
+0x4B176459 0x61DAAA46 0x4D00A9DC 0xD672D16E
+0x90C8467F 0xDE43C68A 0x8861312A 0x4DDDC927
+0xDD321D0A 0x0B97424B 0x29069CC3 0x6B27A5A1
+0x6A3D3DFE 0xDD73C6ED 0x1D39EC83 0x6B48286F
+0x42E1F18E 0x71277EA1 0xD93A81A4 0xBA29AB3A
diff --git a/tools/wtp/keys/CSKA.txt b/tools/wtp/keys/CSKA.txt
new file mode 100644
index 0000000..bf0ba15
--- /dev/null
+++ b/tools/wtp/keys/CSKA.txt
@@ -0,0 +1,55 @@
+Key ID: CSKA
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000007 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x11E8C5D7 0xA7284A13 0x3C4F173C 0xF4674A70
+0x49EDC183 0xBBF0CA6F 0x851479C3 0x3D241D58
+0x02DFF511 0x77A29190 0xB69846C0 0x62E1C683
+0x558312B9 0xD2085079 0xFEA9829D 0xEB8E2947
+0xE92D664C 0xC9404EE4 0xDF0231BE 0x61C1426B
+0x91D7FCC2 0x9466DA99 0xF9E5EBD1 0xD362C8A7
+0x0A3008C5 0x845CB45F 0x8DB7018E 0x9BCA46DF
+0x4B276169 0x2FAF3BD7 0x0C57D378 0xCD0E6AFE
+0x7A16FFF3 0xBB37863A 0x814485D4 0xE8EF0571
+0x003A9A95 0x619A7480 0xB3E64B5C 0xFEF5F3AA
+0x5DC284FF 0xDA21A3F9 0xE32AB325 0x9E4E41AC
+0xEB47DB7D 0x0D064650 0xC0F1DC47 0x324A9164
+0x7A85B4B0 0x911DF41D 0xD2DDEF25 0xBA99F9EC
+0x502321EC 0x2910C2B2 0x95C024A8 0x1CF3BF20
+0xE53D9716 0x054A7676 0xD1AAF3E2 0xF21F50F4
+0x0DE0D6BB 0x40958F61 0xAD65B5E7 0x8FA580E8
+RSA Private Key:
+0xC6EEA61F 0xC860100C 0x4B933434 0x61FAB0F3
+0xA0AAF479 0xCF2997BF 0xFEB11335 0x096321DA
+0x8F8EC581 0xA6CF6C36 0xC1D38FC5 0x36640D6E
+0x2DBEA237 0x50A6D2EF 0xA2540CBC 0x29896CCA
+0x12F55948 0xADAEF6BC 0x2DB2D71A 0xE8E388DD
+0xE23BD7EA 0xA1B0D84E 0x902E8A42 0x62A7AA94
+0xE7F36DB7 0xAF19836E 0xBD3AC4AD 0x95911DC7
+0x25972D96 0x067A33A3 0x64A7B80C 0xE6221AAE
+0xC84C6DB4 0xF62C80E3 0x802E5C42 0x2146B7A2
+0x492CF183 0xC4CCEC12 0xF520E631 0xFFD9FE3C
+0x9FAE12FF 0xB1728523 0x45061997 0x3B2FC03D
+0xB3E5B1A4 0x266E9C54 0x1B90440A 0x2BC1827C
+0x7F37AC19 0xCB969096 0x8BD68FE0 0x63CCDA8F
+0x9DBBE046 0xBCB94062 0x15649785 0xBAFE3FE0
+0x69E43A27 0x49E610EC 0x8BAAB520 0xFE047947
+0x01FB8C63 0x76F0CB57 0xF43319FC 0x14855B8E
diff --git a/tools/wtp/keys/CSKB.txt b/tools/wtp/keys/CSKB.txt
new file mode 100644
index 0000000..6084d05
--- /dev/null
+++ b/tools/wtp/keys/CSKB.txt
@@ -0,0 +1,55 @@
+Key ID: CSKB
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x36C618D9 0xAFE3EE38 0x8AF837F2 0xC887F13B
+0xA745AFF5 0x748BBBDF 0xD785352D 0x62A4757D
+0x81D121A2 0xD50B3AAE 0xE16500FC 0x18648DEA
+0x50DF958C 0xD9FA1B5A 0xFA37B037 0xD9197A35
+0xB97A003F 0xF3A402CA 0x8BFAF70E 0xA2406B13
+0xB9371AC4 0x8E0F3AB3 0x4896D404 0x0B238BFE
+0x6EFEBAA8 0x96F010F8 0xC6D34E79 0xF6DE9A49
+0xB312164F 0xC8119D13 0x0B84C5B6 0x93277CED
+0xD5F9A12D 0xCB3B59CE 0x5322D16C 0xA29D250B
+0x83320C87 0x5542A966 0x035FAC5B 0x5BF5C92F
+0x5E99AB5B 0xD8CDEAC2 0x34595E36 0x9289FA67
+0xBA8D7E72 0x24E07DDC 0x0F977301 0xFFF8161F
+0x107EF118 0x0621738D 0x185284BA 0xEBB0DE71
+0x1A2367A2 0xEE984D0C 0x6533B577 0xACF96802
+0x498C19AF 0xAFA2144E 0x48902B73 0x95FA2FBF
+0x50FAC022 0x6E1B0056 0x27F6D1F8 0x8EF6FA84
+RSA Private Key:
+0xBB1EEA23 0x593266DA 0x7A7B1C92 0x77ABDDE2
+0x42C66A5F 0x0C6144AB 0x7357216F 0x59743CB4
+0x2060382C 0xA38AD06D 0x4EC9B657 0x03175B62
+0xCB84C854 0xA9FD919F 0x0C3C005C 0x5F68F82C
+0x8AC5081E 0x667FD97B 0x5C8C056E 0x2B8519DE
+0xD145E538 0xA3480B41 0x01BEE99C 0x84974DD9
+0x1D58B9C8 0xA91E41DC 0x7795164E 0x2D3B44D4
+0xBE000589 0x8D6914C3 0x89395A85 0x0A2EE3F3
+0xE3FBC0C8 0x877CE689 0xE2173648 0x6C68C35C
+0xACCC085A 0x38D71B99 0x57951D92 0x3D4E861F
+0xE9BBC792 0x3B33F1D6 0x22E63ECF 0x61B1519A
+0x7C5E544C 0xC34053E8 0xB50FA200 0xAAA56414
+0xB5A9F610 0x04164D08 0x658C587C 0x9D20944B
+0xBC179A6C 0xF46588B2 0x4377CE4F 0x1DFB9AAC
+0x865D6675 0xCA6C0D89 0x30601CF7 0xB951752A
+0xE0A72AC1 0x4967558E 0x6FF9E150 0x5F4F51AD
diff --git a/tools/wtp/keys/CSKC.txt b/tools/wtp/keys/CSKC.txt
new file mode 100644
index 0000000..a47d9c7
--- /dev/null
+++ b/tools/wtp/keys/CSKC.txt
@@ -0,0 +1,55 @@
+Key ID: CSKC
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x8F2CCDA1 0xF0763D8D 0x289797F6 0x52A50EB7
+0xBE0B7819 0xB16DB7D9 0x2E41F5AC 0x0DAA1585
+0x1E0CF747 0x8CB020D0 0x93A82B06 0x6EAE42A7
+0x63BF0E85 0x4151686D 0x7BA4D30D 0xE6C11356
+0x5A19BBDA 0xB989FA3F 0x76AA5532 0x12C8CDB4
+0x8F789E57 0xA239EE27 0xDAADBE2C 0x5B53D1D0
+0xECED3963 0xBA17B643 0xF7EFC874 0xC4C54C13
+0xD76A5EB9 0x64DD76BF 0x5F73A653 0x3ED4AA3B
+0xE67990C3 0x51207475 0xD261F849 0xE9200580
+0x6C28C989 0x7EECBDD6 0xC0080FB2 0x895F54A7
+0x26C1E5C1 0x88DA7340 0xA455B55B 0x6A07B30D
+0xB8187B14 0x9FBCA233 0xEBC045E4 0xC2976B5F
+0xCEDE2508 0x0D9657C5 0x6AE2C65A 0x7C5BAFCE
+0x1C7B5F3B 0xDC266B86 0x4569593A 0x6EE56793
+0x6CBD166E 0x596D7B05 0xB7E89B0E 0x4137FE40
+0xE1A6D41B 0x1EEBC811 0x51713716 0xA88C811D
+RSA Private Key:
+0xD2BACB2B 0xA705FA10 0x577748CA 0x52EE63AA
+0xD2C8BC86 0x6D1AAD20 0x02FE89A7 0x44D85910
+0xA5CFC1D9 0x0DB20570 0x33A091D0 0x9C987EC7
+0x16BFC2B1 0xF7DFAD93 0xFB3C6FD1 0x56BED3A5
+0x0BEFEB87 0x137B0871 0x4B8D9717 0xC7FE8455
+0xD510C684 0x99E8F858 0xD49E92F0 0x8BD8B5D2
+0xCF1C7A75 0x95927BDB 0xC3F3F1F4 0xD7DE7AF6
+0x5F9D0AD6 0x48C71D50 0xEE7186A9 0xBEC9F02A
+0x44510B2B 0x3615A2F9 0xE196A586 0x9B6AAE55
+0x481B3106 0x549DD3E4 0x80055FCC 0x5B94E31A
+0xC48143D6 0xB091A22A 0x6D8E78E7 0xF15A775E
+0x2565A762 0xBFD316CD 0x47D583ED 0x81BA4795
+0x349418B0 0xB3B98FD9 0x9C972EE6 0x52E7CA89
+0xBDA794D2 0x92C447AE 0x2E463B7C 0x9F439A62
+0x9DD36449 0x3B9E5203 0x2545BCB4 0xD625542B
+0x4119E2BC 0x69F28561 0xE0F624B9 0x705DAB68
diff --git a/tools/wtp/keys/CSKD.txt b/tools/wtp/keys/CSKD.txt
new file mode 100644
index 0000000..fe1df49
--- /dev/null
+++ b/tools/wtp/keys/CSKD.txt
@@ -0,0 +1,55 @@
+Key ID: CSKD
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x6EB085A9 0x8488C1C8 0xA81FFD8D 0x9DC499D6
+0xC34C208D 0xCFAA69C1 0x202C0461 0x4E45F6B1
+0x8F844929 0x309E290D 0x5BC24AF3 0x960C80C1
+0xFDF83333 0x720486A8 0x80AD6193 0xC7D201AC
+0xCA08C74A 0x38B1E1C7 0xEAD311E7 0x91E97BCD
+0x2C085CC2 0x57C4FC83 0x6C666299 0x666EF86C
+0x5E80A8EE 0x90F3BB25 0x6277151C 0x9098AB63
+0x127B43E5 0xB6208EEC 0x0B895168 0xC65788D6
+0x4067BD6C 0xCD10D4B3 0xA689BA69 0x3D5C3F86
+0xB9E7694E 0x568672CB 0x4EF93A0F 0x73471E5C
+0x121CBB41 0x26775A98 0x33D278EE 0xA92202EE
+0x6EB4788C 0x8B92D176 0xB6A79E04 0x655C272A
+0xE09264D2 0xEB7575CF 0x66272B8C 0x4452A007
+0xDA070D6A 0x0912A1D6 0x7AC1D664 0x636AA549
+0x9CB5D1B6 0x47EAE2B0 0x80ECD81C 0xA29C3099
+0x034C3C2A 0xAAEC7AD9 0xD5750A6A 0xB339BB8B
+RSA Private Key:
+0xB9CB7E99 0x0F71C020 0xC1D79528 0x7FC5EE58
+0x0E0CA936 0xE6E2362E 0x4B473DAA 0x0811DB3B
+0x6D1C2821 0x1063FCFB 0xA4AB5680 0x09CC4216
+0xE90A55CF 0x1C3F847B 0xAA65FAAA 0xBB7D08C7
+0x348A14B1 0xDC0D0627 0xF3D84C49 0x35147D4D
+0x24D81913 0x5E69404F 0x5D7F3E03 0xFA48621D
+0x04964899 0x3B951031 0xAE410772 0x85175DDD
+0xD68F954F 0x319A1804 0x9A039F2B 0x0DD7A083
+0x59D7D80D 0x7B0A1938 0xFD85D63F 0xBE6A8C83
+0xA2BE0BFB 0x33EA44E0 0x2F625609 0x785DDF04
+0x7144705A 0x1714698E 0xB8B17BC2 0x32479B5B
+0x0F391521 0xED581747 0x6D97C535 0xD66A7DE6
+0x2057D617 0x8D4679E3 0xD6E44D54 0x8F64C66A
+0x82D1080C 0xD23E611A 0xB00DE708 0xD53FFCC5
+0x2AD37DD3 0xC4C0219D 0x1A27B4DD 0xC7F7505C
+0xCEC7574C 0x335AB01B 0xB3463973 0x6B890A20
diff --git a/tools/wtp/keys/CSKE.txt b/tools/wtp/keys/CSKE.txt
new file mode 100644
index 0000000..108cc19
--- /dev/null
+++ b/tools/wtp/keys/CSKE.txt
@@ -0,0 +1,55 @@
+Key ID: CSKE
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x1B9BA633 0x20E1A628 0xF590848D 0x61EEE18A
+0x818E9C11 0xDB41F99E 0x60C41E9A 0x0D3F2F12
+0x25CFF022 0x40D6F206 0xBD956736 0x0CE10AF5
+0x9644161D 0x4E311CED 0xC759D55C 0xF6067A34
+0x14409A8E 0x172FB96C 0xE4DF02F1 0x9A823EA9
+0xAE011A2C 0xB0193083 0x569C00B8 0x947AEC1E
+0x1CC0A4EB 0x4EF3D0CA 0x8B618734 0xEA825D95
+0x01E741B8 0xD819A62F 0xD69C38CA 0xEC0CFD96
+0x4714DDEC 0xECD1BA71 0xDD1534FB 0x12400336
+0xB820B99C 0x5E483F3A 0xB691E5DD 0x3A65E34F
+0x8068C6D7 0x6C67F719 0x9734CB95 0x3B69268C
+0x69028406 0xC782F8D9 0xF344CC56 0xFD4696F2
+0xF3D4B293 0x50C07A75 0x419DCC90 0x7EDF9C9B
+0xDA146BF2 0x13BA7DA5 0x5B1ABC21 0xF60EC64F
+0xECC0D7ED 0xB09192C6 0xDCF4ABA5 0xC34DE401
+0x9AC70C66 0xCD16209E 0x5F2C6F9F 0xB0C05D75
+RSA Private Key:
+0xC763DB7B 0xF6C91208 0x18040603 0xFECEFF13
+0x3A0A1E66 0x4B651EA9 0x80155164 0xC4D716B2
+0x9CAE246E 0x22EC05D9 0x79B8EDCE 0xC119E193
+0x79E3EA99 0x9A96DE1E 0xDF9DF246 0xD062612B
+0x338D7445 0x6D166B4E 0x5376C7C7 0x14BE72FF
+0x9AD8CDBE 0x36DF86C3 0x33057DF7 0xC0B81A2A
+0xBFAAC360 0xC08E885B 0xB0AEF3E7 0x1E70A7DE
+0x76E5BF0B 0x6F0AE9CC 0x3101DBE0 0x7D69C660
+0xDA0DE947 0x488BD1A0 0xE8B8CDFD 0xB6D55779
+0xD015D112 0x3EDAD4D1 0x79B6993E 0x26EE978A
+0x559B2F3A 0x48454F66 0x64CDDD0E 0x279B6F08
+0xF0AC5804 0xDA575090 0x4CD88839 0xA8D9B9F7
+0xA28DCC62 0x8B2AFC4E 0x8113DDB5 0xA9EA6867
+0x9162F2A1 0x0D26FE6E 0xE7672816 0x4EB48434
+0x9DD5E549 0x20610C84 0xE8A31D19 0xD78942AB
+0x672F5D99 0x88B96B14 0x94C84A6A 0x75D593A3
diff --git a/tools/wtp/keys/CSKF.txt b/tools/wtp/keys/CSKF.txt
new file mode 100644
index 0000000..401bb14
--- /dev/null
+++ b/tools/wtp/keys/CSKF.txt
@@ -0,0 +1,55 @@
+Key ID: CSKF
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000007 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x8BAAF09B 0x9F054D65 0xD96E8251 0xD6A54983
+0xE298F2A7 0xC3A61186 0x1569AF1C 0xCF08DF01
+0xDA22D992 0x464A7549 0xA0743F28 0x21F21FDB
+0x62883E78 0xF273C89E 0x398BA51A 0xA29A01CB
+0x1D1DF0BB 0x55D8CEEF 0xDD2F3725 0xAF498E7D
+0x1757D055 0xEF607138 0x8F576E6D 0x3986C646
+0x0D001EB0 0xD2E72201 0x24A7CEC7 0xBA1B007D
+0xF0377102 0xB913E5F2 0xA485BE45 0xB9EB9F8D
+0x316D5671 0xA8E01324 0x5B4EA32C 0x576D3154
+0xB6DB746D 0x25B9473C 0xEC73B697 0x1663B961
+0x942F4E3E 0x8F18117B 0xC37B494C 0xC1F94666
+0x822AAFFE 0x3C5456AE 0xB3691FA5 0x7299BB72
+0xE1FBD6A6 0x24C49D7F 0x36BB9FE6 0x0BE04D0F
+0x4D24A7ED 0x7F0E149F 0x015F6836 0xFCD94761
+0x70CEC997 0xCA69DC43 0x0ECF7082 0x067CB240
+0x5163D25E 0x506BB33B 0x632BCC5F 0xA56F6AE8
+RSA Private Key:
+0x456DA927 0x6198978D 0x8E44F7E8 0x4B7B37A6
+0x625789BE 0x8F399FA4 0x7386EA30 0x157F36A6
+0x9B609CDE 0x4C90695D 0xC5511506 0x913D6E4D
+0x0311105D 0x050568A0 0xB33E1644 0xE135FF5F
+0x44B49B50 0xE4D448F7 0x4CBFEEC6 0x7EA4B074
+0x6B570771 0x3FA3EB32 0xB3D8AE4A 0x3634DAA7
+0xD7E03D66 0xF9431503 0x050093E2 0xFDC7F101
+0xD8A7D844 0xE0765B76 0x9D65513D 0x5EADA530
+0xDA2986E2 0xE656E919 0xF813991F 0xF54DFEA9
+0xA72F0A04 0x88A8E9BD 0x169BCB90 0xEB6BCD8F
+0x20B41350 0x1D11310F 0x427CA212 0x1CD6A000
+0xEF430FFF 0x98CE870E 0xC94B169A 0x766DCF08
+0x58462B9B 0x87FA275B 0x703CDFED 0x7632802F
+0x5BAC77F2 0xC877C596 0x934425DD 0x9008E9D7
+0x2C016B6C 0xD9B95430 0x784B0738 0x7259119B
+0x3A22BAD5 0x14DF36E1 0x46D62444 0x762B0338
diff --git a/tools/wtp/keys/KAK.txt b/tools/wtp/keys/KAK.txt
new file mode 100644
index 0000000..2c38955
--- /dev/null
+++ b/tools/wtp/keys/KAK.txt
@@ -0,0 +1,55 @@
+Key ID: DSIG
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size in bits: 2048
+RSA Public Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x77F56E6F 0xF5DB0AD4 0xEB994D35 0xA863FE85
+0x747D27ED 0x9EE303BA 0xFB55D06C 0x8DBAEF04
+0x9AC8F88F 0x8867FA62 0x8DFA5127 0x286D8EA5
+0x27FF8E37 0x70352983 0x8C91B233 0xF405C5D9
+0xA6A5C82C 0x837A1F87 0xBA48B0C1 0x93F53C48
+0x28D7FC8A 0xF9C7ADA9 0x670C502A 0x1FC12DA2
+0x71731332 0x0EB4C539 0x302C6048 0xF509D000
+0x76652F8E 0x934F1993 0x1FA1B34E 0xD16CB85A
+0x158676A5 0xE7C149B0 0xC205E905 0x7E93ABD2
+0x8EED7C51 0xCBB4B5DB 0x46B79B5D 0xFE93923A
+0x19880137 0xA3B4AD2E 0x0F1052D8 0xA603C887
+0x574BB1E8 0xE9BCA9D7 0xB895700B 0x56D44368
+0x8A98798E 0xFB40038D 0xB6486D0F 0x5328EC7D
+0x888616A8 0xBCD39C2C 0x90274217 0x0D902683
+0x2E195296 0x661D62BC 0x278042A2 0x30ECAD89
+0x6A2541EF 0xA94696C9 0x21482342 0x8B5E9CE9
+RSA Private Key:
+0xD4AFB845 0x882291D2 0x0AEE53EB 0xF1A487FB
+0xD4C8B4A3 0x62FF05F1 0x316CB1CE 0x88C4E130
+0xA0DFD69F 0xBBB22559 0xB8EE1753 0xB61D13D3
+0xB2A0B791 0x83EFABDF 0x28E75756 0x6DFBF10E
+0xC4022294 0x403D1E65 0x977AE60F 0xF1286EF7
+0x77881617 0xCD439140 0x805117C3 0x8CA829F3
+0x29BBC428 0x170A0A39 0xD10CD562 0x0B156871
+0xDB551DE2 0xFFE0D58E 0xE7FEDD0E 0x77037604
+0x044E17BA 0x94C041F0 0x26CDFB67 0x4C83EF2A
+0x82FC4C10 0xF58A8AC5 0xA7BE5245 0xCC83EA0B
+0x384E66A4 0xED8A8909 0x030343C4 0x5467281B
+0x117589FB 0x2EBF552B 0xF1B77CCF 0x115DA714
+0x821E7EB6 0xCBD99A4F 0x2474E29C 0x43D4FC19
+0x4E8137BB 0x8C2A526F 0x1CD4A6D1 0xCF833AE7
+0x3C6B76EA 0xE13913BF 0xD4B34086 0x09C8EF81
+0xAED44063 0xBB74848E 0x6D0E6D73 0x1BDFB8FB
diff --git a/tools/wtp/keys/KeyGeneration.txt b/tools/wtp/keys/KeyGeneration.txt
new file mode 100644
index 0000000..097f520
--- /dev/null
+++ b/tools/wtp/keys/KeyGeneration.txt
@@ -0,0 +1,7 @@
+Seed: ArmadLP Trusted platform
+Key ID: DSIG
+Encrypt Algorithm ID: 7
+Hash Algorithm ID: 32
+Key Size: 2048
+Output Binary Filename: ./KAK.bin
+Output Text Filename: ./KAK.txt
diff --git a/tools/wtp/u-boot-ntim.txt b/tools/wtp/u-boot-ntim.txt
index a9b958a..20ace46 100644
--- a/tools/wtp/u-boot-ntim.txt
+++ b/tools/wtp/u-boot-ntim.txt
@@ -1,44 +1,42 @@
-; Revised by MBU version: 0x03031108  MBU Date: 0x20141028  File Written: Wed Mar 25 14:07:19 2015
-; Created by MBU version: 0x03031108  MBU Date: 0x20141028  File Written: Thu Mar 05 16:37:42 2015
-Version:                     0x00030500
-Trusted:                     0x00000000
-Issue Date:                  0x02272015			; today's date
-OEM UniqueID:                0x21796B53			; Sky!
-Processor Type:              <undefined>
-Boot Flash Signature:        0x5350490A			; Select SPI'0A
-Number of Images:            4
-Size of Reserved in bytes:   44
+Version:                       0x00030500
+Trusted:                       0x00000000
+Issue Date:                    0x12162015		; today's date
+OEM UniqueID:                  0x4D52564C		; MRVL
+Processor Type:                <undefined>		; ARMADALP
+Boot Flash Signature:          0x5350490A		; Select SPI'0A
+Number of Images:              4
+Size of Reserved in bytes:     44
 
-Image ID:                   0x54494D48				; TIMH
-Next Image ID:              0x4F424d49				; OBMI next
-Flash Entry Address:        0x00000000				; TIM flash addr
-Load Address:               0x20006000				; TIM ISRAM addr
-Image Size To CRC in bytes: 0x00000000
-Partition Number:           0x00000000
+Image ID:                       0x54494D48		; TIMH
+Next Image ID:                  0x4F424d49		; OBMI next
+Flash Entry Address:            0x00000000		; TIM flash addr
+Load Address:                   0x20006000		; TIM ISRAM addr
+Image Size To CRC in bytes:     0x00000000
+Partition Number:               0x00000000
 Image Filename: TIM_UBOOT_NON_TRUSTED.bin
 
-Image ID:                   0x4F424d49				; OBMI - 64K code + 16K data
-Next Image ID:              0x57544d49				; WTMI next
-Flash Entry Address:        0x00004000				; OBMI flash addr
-Load Address:               0x1FFF0000				; OBMI RAM addr
-Image Size To CRC in bytes: 0x00000000
-Partition Number:           0x00000000
+Image ID:                       0x4F424d49		; OBMI - 64K code + 16K data
+Next Image ID:                  0x57544d49		; WTMI next
+Flash Entry Address:            0x00004000		; OBMI flash addr
+Load Address:                   0x1FFF0000		; OBMI RAM addr
+Image Size To CRC in bytes:     0x00000000
+Partition Number:               0x00000000
 Image Filename: spl/u-boot-spl.bin
 
-Image ID:                   0x57544d49				; WTMI
-Next Image ID:              0x4F534C4F				; OSLO next
-Flash Entry Address:        0x00024000				; WTMI flash addr
-Load Address:               0x200040F0				; WTMI RAM addr
-Image Size To CRC in bytes: 0x00000000
-Partition Number:           0x00000000
+Image ID:                       0x57544d49		; WTMI
+Next Image ID:                  0x4F534C4F		; OSLO next
+Flash Entry Address:            0x00024000		; WTMI flash addr
+Load Address:                   0x200040F0		; WTMI RAM addr
+Image Size To CRC in bytes:     0x00000000
+Partition Number:               0x00000000
 Image Filename: tools/wtp/wtmi.dummy
 
-Image ID:                   0x4F534C4F				; OSLO
-Next Image ID:              0xFFFFFFFF				; NONE
-Flash Entry Address:        0x00026000				; OSLO flash addr
-Load Address:               0x60000000				; OSLO RAM addr
-Image Size To CRC in bytes: 0x00000000
-Partition Number:           0x00000000
+Image ID:                       0x4F534C4F		; OSLO
+Next Image ID:                  0xFFFFFFFF		; NONE
+Flash Entry Address:            0x00026000		; OSLO flash addr
+Load Address:                   0x60000000		; OSLO RAM addr
+Image Size To CRC in bytes:     0x00000000
+Partition Number:               0x00000000
 Image Filename: u-boot.bin
 
 Reserved Data:
diff --git a/tools/wtp/u-boot-tim.txt b/tools/wtp/u-boot-tim.txt
new file mode 100644
index 0000000..b2559b0
--- /dev/null
+++ b/tools/wtp/u-boot-tim.txt
@@ -0,0 +1,836 @@
+Version:                        0x00030600
+Trusted:                        0x00000001
+Issue Date:                     0x12162015		; today's date
+OEM UniqueID:                   0x4D52564C		; MRVL
+Stepping:                       0x00000000
+Processor Type:                 <undefined>		; ARMADALP
+Boot Flash Signature:           0x5350490A		; Select SPI'0A
+Number of Images:               1
+Number of Keys:                 16			; CSK0-CSKF Keys. DSIG KAK key is not counted
+Size of Reserved in bytes:      48
+
+Image ID:                       0x54494D48		; TIMH
+Next Image ID:                  0xFFFFFFFF		; No next
+Flash Entry Address:            0x00000000		; TIM flash addr
+Load Address:                   0x20006000		; TIM ISRAM addr
+Image Size To Hash in bytes:    0xFFFFFFFF		; Hash complete image file
+Hash Algorithm ID:              32			; SHA-256
+Partition Number:               0x00000000
+Encrypt Algorithm ID:           0x00000000
+Encrypt Start Offset:           0x00000000
+Encrypt Size:                   0x00000000
+Image Filename: TIM_UBOOT_TRUSTED.bin
+
+Key ID:                          0x43534B30		; CSK0
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x84724B4D 0xF58CA3E8 0xD379F131 0x1ABFF997
+0xC8D58E4C 0x5CA18F30 0xB739BA64 0x5938611F
+0x8EE5D2B6 0x0EF040E8 0x153DB054 0x1A22F8BE
+0xCBB9BCEF 0xC2793935 0xF133342F 0xA31B788C
+0x44AEA9C7 0x0F255B5E 0xA8D0DBDB 0xC2CD9DC3
+0x34DCC5A1 0xF5812DEF 0x58C2D51C 0x879562B1
+0x1F2A5A8B 0x4884300D 0x51F9A7DF 0xB8229036
+0xC73EA8B4 0xDDA0AF27 0x9CE80B9F 0x393B0C79
+0x5630EA68 0x558D53AE 0xCCF630EF 0x72247090
+0xBCB40C87 0x3DC04183 0x9FB8E19F 0x7BD96878
+0x534168EA 0x20FA2569 0x135829A3 0x0A34F9BD
+0x5994A70C 0x0D8E43E0 0xE352D564 0x411D3AA5
+0x4F666B2A 0x5CCFFADA 0x0B14C219 0x82E9B28C
+0x1AFDC5A8 0x2E699FCC 0x1D79E540 0x457CE2EB
+0x4D56DACE 0xAA900176 0x366DAFDC 0x1F895984
+0x4EBC5283 0x5668D9E4 0x889D174B 0x8BB67F04
+
+Key ID:                          0x43534B31		; CSK1
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x80EED371 0x0AB09B9B 0xA8BE98CD 0xB3246236
+0x4518BCB8 0x497B8AE2 0xB7B5180C 0x71C9A948
+0x458692E5 0x855AD78F 0x0F118A08 0xCBB5B7F0
+0xB4A915FD 0xD213B4A4 0x6526A4AB 0x7A5765D1
+0x147FA13D 0x544031DB 0xB9F08C82 0x2AA3B954
+0x318DA699 0x2547A92C 0xEBB021A4 0x41B46DFB
+0x420261F2 0x53C2E1E0 0xB396E865 0x21745D50
+0xC85A7C8D 0x9B1953C0 0x06B64867 0xD2033737
+0xAB830B2C 0xF091685F 0x8A8C7F6E 0x7AE0436E
+0xC3FAFD35 0x34A1BC5B 0xBCCE1684 0x2A5B3168
+0x378DBDF4 0x20B5DE97 0x2E11AD52 0xF43E6531
+0x6608353F 0x4D11B545 0x7E9AEA03 0xD1299F13
+0x995C3700 0xEBF5CEDF 0xABDC0164 0x50B8250B
+0x387E5E50 0x62258E08 0x019F98D3 0xC530F7AB
+0x9B120CAF 0x52335B28 0x3B8896A0 0xEB5C6D17
+0xE67F80DB 0xD5A4B203 0x043872B6 0x9BD47764
+
+Key ID:                          0x43534B32		; CSK2
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x0000000B 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0xA909FF45 0xA432D24A 0x1E9A2FE9 0x062A3CC5
+0xC5286B43 0x5F599935 0x5342A93F 0x53BE93E6
+0x978DBC01 0x29F90652 0xFEB50E31 0xA40EFB45
+0x1FD4AC41 0x71C209AD 0x7DA7868D 0x224A835F
+0x4E8B3F95 0xD4D60A04 0x58CFB7F2 0x8861C7D3
+0x87704D6B 0x4824ABDB 0x92020A1D 0x8F08D3DF
+0xC1A3A6F5 0xE200120C 0xBFEE8322 0xB79FF73E
+0x8DEFFF8E 0x3B49DC35 0x4B7530BA 0xBE1DD02E
+0x76743693 0x11FE0B29 0x98E82CD5 0x324A8FC6
+0xDA1685C8 0x4E1F9224 0x75488EAB 0x35B0A53F
+0xEDAD6DB0 0xB9ECF27E 0xB8ACDB84 0xBEEA13D5
+0x1C160A24 0x6AC615F3 0xEC5BBC36 0x09F04D5E
+0x4BEA748E 0x75A8BA08 0xA1C096CF 0x301FFB75
+0xE2DE156F 0xFA694BE8 0x41EC3BC2 0xA52AEB45
+0x0271D5B2 0xFBBF692B 0x4236E95D 0x0E105EA1
+0x82E6A06C 0xEDE2FA67 0xAB731C64 0xD380DE78
+
+Key ID:                          0x43534B33		; CSK3
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0xCC14894B 0x1D54EFAB 0x2C006ECC 0xECBA39B5
+0x386B9504 0xE5127FA9 0xBBAF3A06 0xB351C24B
+0x90A1B218 0xDB877D1E 0xFAAECD8C 0x2ABDC223
+0x88AB687E 0x3BB63A7A 0xE2CECAA9 0x9F51C0D8
+0xF55351ED 0x68C4C09E 0xF4FED718 0x111E24B9
+0x8D938BEB 0x8ECF27F9 0xBBE7D8D5 0x1AC4F691
+0xD54FBDB1 0x18A6D6BE 0x9619F30B 0x96461657
+0xDF2961EC 0x3AE58790 0x88ECC5BD 0x181A4B15
+0xED03EB8E 0xB365ED3E 0x7C95DCEA 0xE13FEF7A
+0x046073C2 0xDF910B26 0x3F629BFD 0xDA8AD3F3
+0xE3FBCFD8 0x49338817 0x8BBF17BD 0x02AAEAFA
+0x4678A53E 0xB7D51770 0xBF853DCA 0x75487D2D
+0xD57ECF69 0x9E444CF0 0x0B7B1D0D 0x38C1DEE9
+0xEDC2BDB1 0xDF03CEA1 0xDDD31093 0x0317A73D
+0x02EB8F74 0x5519D9F9 0xA3501FDE 0xC4BA749D
+0xA82D50A6 0x0E42C2C6 0x0E56FC52 0xA533A925
+
+Key ID:                          0x43534B34		; CSK4
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x1CE759B9 0x4AFF4832 0x9CFA2713 0xCD988395
+0xCE44E7B2 0x1BEDD8C1 0xFBB53605 0xA042C260
+0x9BE6075A 0x02E3372E 0x3E1DAE9C 0xAB6C0387
+0xFE6C24F8 0x7335FEE3 0xB43A2AC4 0xE7BB7C7D
+0x388349C2 0xFD5671CE 0xA8364768 0x5DC26EE9
+0x0B3DFDAE 0xD803592A 0x26AB97BF 0xE3EB4EBB
+0x4CB562E4 0xAB5FB74E 0x3F200C72 0xE569DF3E
+0xB310E40C 0xA7E87BCD 0xEE99B2F1 0xAE69FFF9
+0xD47A88E7 0xB0EBEB19 0x21D613DA 0xF84E7A42
+0x308C1F39 0x0A3C4E8D 0x83977B36 0xBE02DA3E
+0xE36AF13A 0x098C57C5 0x5867DE7B 0x64FA0437
+0xA5CD852B 0xEC154C71 0x5073B641 0x231BFCD4
+0x05EEDE28 0xF64E1471 0x6EA07407 0x7D0430ED
+0xD900FC06 0x288FC558 0xB2A95E0D 0x451A2DE7
+0x5E51F306 0x3F3E6FA2 0x7338E97A 0xC4BAA7BB
+0x0C4A7ED5 0x41842DCF 0x03E84F51 0x981B690A
+
+Key ID:                          0x43534B35		; CSK5
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x0000000B 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0xB42F42BD 0x2D6BB77C 0x3DBBF682 0x97E79D79
+0x112F41CD 0x1440BA6E 0x72B8CBA1 0x9F3AC71C
+0xAFFB5B83 0xCF6CE01A 0xA451F1C4 0x840F8B39
+0x47F5B6B4 0x52D24D6B 0xF940FCE5 0x938714CD
+0xAE8ED49A 0x011C7F4D 0x20A0EAB3 0x854433C6
+0xFD25B036 0x5F402E71 0x542B5679 0xCACC93CB
+0x1DDB8ADC 0x426B1636 0x99944330 0xF8F9CEDF
+0x013B1D84 0x9AB89FF3 0x025DC47D 0xF092EF75
+0x8E61250C 0x6C07157B 0x60F43C83 0x29936A4E
+0x6BF5A125 0x5B55EAEC 0x15828E0C 0x48E23DF3
+0xFB41A5E8 0xC48C9CB4 0x217FDBEC 0x837D569B
+0x859FEB98 0x249012F9 0xCC50C508 0xA6253F45
+0x236BC1A8 0xA7B3062F 0x944A9B4E 0x82D973E2
+0x7C0BC74E 0x38BCAEE8 0xF59EBEE0 0x64376CD6
+0xB253A4D0 0xCFCF01A8 0x073FB688 0x27B3ED9B
+0x94C016CC 0x0783D671 0x6FD68A39 0x8869DD7C
+
+Key ID:                          0x43534B36		; CSK6
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x3F236983 0xEBDB683B 0xDDA3648F 0x914B5800
+0x31B22E27 0x174F42D6 0x9FCF6B74 0x702CECF7
+0x64DF7FBA 0x402FDB7F 0x9F508CBD 0xA297B410
+0x0BDA7224 0x8531ADDE 0x5B3B720A 0x888138ED
+0xD12A0E97 0xFDA8AF45 0x4EC9073D 0xEBB56CEF
+0xFD1CB219 0xDE792A64 0x8D2B323B 0x1853D025
+0x789D2E0E 0x14546422 0x856E4789 0xF4858605
+0x3EC6F74A 0x8E6E0FDA 0x591A6D47 0xAD05DC1A
+0xA50D3F8D 0x8B98A196 0x08434948 0x767F9DE6
+0x2D7A6D73 0xF7E5EED0 0x53A46A03 0x008457A2
+0x4F086954 0xC98D0C87 0x37C60DD9 0x5A3AF863
+0x8B535028 0xAF3FC78E 0x0AC0CB91 0x63BD1E08
+0x0E9013FC 0xC86D4C3C 0xE0B434F5 0x448790F3
+0x535EE091 0xD2C23186 0xEDAF39A9 0x16AF5681
+0x46DB7772 0x5F7EC0E6 0x0A96AF66 0x432713A5
+0xD813F959 0xB874402E 0x3A4A2145 0x86826C46
+
+Key ID:                          0x43534B37		; CSK7
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x8FA6AA23 0xCCF84CA4 0xFEB69002 0x996E098C
+0xB909AE3D 0x2D9DF389 0x7FD369A3 0x7FF61B61
+0x4D8E4730 0xEC43162D 0x2184D5F8 0xA05A0529
+0xAECB5125 0x4E59A548 0xE9247D4C 0x0053218C
+0x2F5084E0 0x8CD04BA5 0xACE14EE3 0xA24BA650
+0x9492DE2F 0xE76E7164 0x58D84806 0xAB0DEC2E
+0xAF39B139 0xB585F725 0xFF9B8A42 0x46F3F169
+0xE22570CE 0x6681864E 0xF2713D5E 0x7FFFB5AC
+0x8C2E22F7 0xB8BECCF8 0xB80AC29D 0xD8A61BD5
+0x4B51BC1B 0xF5895F18 0x6E0C64C8 0x356CADDD
+0x973136D7 0x0B523327 0x9252B3C7 0x537E92A1
+0x79FC3EC6 0xD7A627D1 0xF6999E76 0xB8FEB250
+0x40DD0A0E 0xF70AD291 0x6D6CE402 0xD3EDC239
+0xA6BCB44D 0xDFDD1C0C 0x0FC00C2F 0x015FD8A4
+0x441669BB 0xD63FA75F 0x6106EA2E 0xC5C433BC
+0x34DE89D1 0xD7360B9E 0xCD3CC15A 0xBD689D5F
+
+Key ID:                          0x43534B38		; CSK8
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x6EB96073 0xD83975B3 0x6BB3FA48 0x77F435AB
+0x1CB0C1DC 0x7B9C076D 0x704A7D7B 0x1DBF8D59
+0x200DB680 0x420439A8 0x33F7BBBC 0x2CB397DB
+0x08FA9EBB 0x2EA446DB 0x46AAA79D 0x5B527159
+0xFA9CC944 0x8CE39B34 0x6E80DF9F 0x4DCA4310
+0x3CC2DD75 0x262A2F72 0xAB74CB32 0xE1153DD3
+0xEE966E87 0x09DBD071 0x0156F389 0x7EDB7DD4
+0xFBA01311 0xD1805968 0xCBE3BFF7 0x7ADFAD6E
+0x77CBCCEA 0x7C89F4F8 0x82259AFA 0xEE472314
+0xEFB72065 0xC32E1049 0x43EEE9F6 0x372E5004
+0x4613B37C 0x3B09569A 0x5588FBDE 0x93979FEA
+0x70485A1D 0x2BC862FC 0x0A970E72 0x33409DE5
+0x4B6C97AC 0x88204426 0x2EDB395F 0x9C689899
+0x64126A7F 0xAC9A4CF7 0x126A2041 0x404AB206
+0x64B85FD9 0xA7BD445A 0x12CA7C28 0x4955918E
+0x40B8BF7D 0x45CB59F2 0xE5A8A467 0x9F94C876
+
+Key ID:                          0x43534B39		; CSK9
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x1A391637 0xF3981418 0x50F8BA85 0xE6765B83
+0x8E69891F 0x8DB23893 0xEE0FCA94 0x7ED0D5B0
+0xA491F91A 0x9B5E8196 0xD9344934 0xEBE12AD5
+0x81D1C27E 0xB20BA288 0x04963930 0x0E7A293B
+0xE7D56DCB 0x8F65E162 0xF6BBDEE0 0x86AEA562
+0x3D08C23C 0x768EA975 0x377D71CC 0x17B9BC0B
+0x6AB327C3 0x407B7862 0x24FF1095 0xFDB888D9
+0xC422FB8B 0xB87FDE7C 0x05D83774 0x1AA1ACB6
+0xAAA7B491 0xF054D003 0xAB637711 0x1FD7D1BE
+0xE9966138 0x077024D3 0xE4E8BBE8 0x2E962E07
+0xECF5A346 0xF39676A2 0xE702A229 0xDBF8B6F5
+0xDDDD3D6F 0x7A5154D7 0xE040D453 0xCC0F85C9
+0x34FA581F 0x95D4B82D 0x6A797D75 0xE1553B71
+0xD47EA44C 0xCE7D12DE 0x734843F3 0x05F18F09
+0xC4CC8D7E 0xD4D0B8A8 0xE48867A4 0x061A328A
+0x939A6DF2 0x8D715E49 0x8F89220D 0xE8B41609
+
+Key ID:                          0x43534B41		; CSKA
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000007 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x11E8C5D7 0xA7284A13 0x3C4F173C 0xF4674A70
+0x49EDC183 0xBBF0CA6F 0x851479C3 0x3D241D58
+0x02DFF511 0x77A29190 0xB69846C0 0x62E1C683
+0x558312B9 0xD2085079 0xFEA9829D 0xEB8E2947
+0xE92D664C 0xC9404EE4 0xDF0231BE 0x61C1426B
+0x91D7FCC2 0x9466DA99 0xF9E5EBD1 0xD362C8A7
+0x0A3008C5 0x845CB45F 0x8DB7018E 0x9BCA46DF
+0x4B276169 0x2FAF3BD7 0x0C57D378 0xCD0E6AFE
+0x7A16FFF3 0xBB37863A 0x814485D4 0xE8EF0571
+0x003A9A95 0x619A7480 0xB3E64B5C 0xFEF5F3AA
+0x5DC284FF 0xDA21A3F9 0xE32AB325 0x9E4E41AC
+0xEB47DB7D 0x0D064650 0xC0F1DC47 0x324A9164
+0x7A85B4B0 0x911DF41D 0xD2DDEF25 0xBA99F9EC
+0x502321EC 0x2910C2B2 0x95C024A8 0x1CF3BF20
+0xE53D9716 0x054A7676 0xD1AAF3E2 0xF21F50F4
+0x0DE0D6BB 0x40958F61 0xAD65B5E7 0x8FA580E8
+
+Key ID:                          0x43534B42		; CSKB
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x36C618D9 0xAFE3EE38 0x8AF837F2 0xC887F13B
+0xA745AFF5 0x748BBBDF 0xD785352D 0x62A4757D
+0x81D121A2 0xD50B3AAE 0xE16500FC 0x18648DEA
+0x50DF958C 0xD9FA1B5A 0xFA37B037 0xD9197A35
+0xB97A003F 0xF3A402CA 0x8BFAF70E 0xA2406B13
+0xB9371AC4 0x8E0F3AB3 0x4896D404 0x0B238BFE
+0x6EFEBAA8 0x96F010F8 0xC6D34E79 0xF6DE9A49
+0xB312164F 0xC8119D13 0x0B84C5B6 0x93277CED
+0xD5F9A12D 0xCB3B59CE 0x5322D16C 0xA29D250B
+0x83320C87 0x5542A966 0x035FAC5B 0x5BF5C92F
+0x5E99AB5B 0xD8CDEAC2 0x34595E36 0x9289FA67
+0xBA8D7E72 0x24E07DDC 0x0F977301 0xFFF8161F
+0x107EF118 0x0621738D 0x185284BA 0xEBB0DE71
+0x1A2367A2 0xEE984D0C 0x6533B577 0xACF96802
+0x498C19AF 0xAFA2144E 0x48902B73 0x95FA2FBF
+0x50FAC022 0x6E1B0056 0x27F6D1F8 0x8EF6FA84
+
+Key ID:                          0x43534B43		; CSKC
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x8F2CCDA1 0xF0763D8D 0x289797F6 0x52A50EB7
+0xBE0B7819 0xB16DB7D9 0x2E41F5AC 0x0DAA1585
+0x1E0CF747 0x8CB020D0 0x93A82B06 0x6EAE42A7
+0x63BF0E85 0x4151686D 0x7BA4D30D 0xE6C11356
+0x5A19BBDA 0xB989FA3F 0x76AA5532 0x12C8CDB4
+0x8F789E57 0xA239EE27 0xDAADBE2C 0x5B53D1D0
+0xECED3963 0xBA17B643 0xF7EFC874 0xC4C54C13
+0xD76A5EB9 0x64DD76BF 0x5F73A653 0x3ED4AA3B
+0xE67990C3 0x51207475 0xD261F849 0xE9200580
+0x6C28C989 0x7EECBDD6 0xC0080FB2 0x895F54A7
+0x26C1E5C1 0x88DA7340 0xA455B55B 0x6A07B30D
+0xB8187B14 0x9FBCA233 0xEBC045E4 0xC2976B5F
+0xCEDE2508 0x0D9657C5 0x6AE2C65A 0x7C5BAFCE
+0x1C7B5F3B 0xDC266B86 0x4569593A 0x6EE56793
+0x6CBD166E 0x596D7B05 0xB7E89B0E 0x4137FE40
+0xE1A6D41B 0x1EEBC811 0x51713716 0xA88C811D
+
+Key ID:                          0x43534B44		; CSKD
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x6EB085A9 0x8488C1C8 0xA81FFD8D 0x9DC499D6
+0xC34C208D 0xCFAA69C1 0x202C0461 0x4E45F6B1
+0x8F844929 0x309E290D 0x5BC24AF3 0x960C80C1
+0xFDF83333 0x720486A8 0x80AD6193 0xC7D201AC
+0xCA08C74A 0x38B1E1C7 0xEAD311E7 0x91E97BCD
+0x2C085CC2 0x57C4FC83 0x6C666299 0x666EF86C
+0x5E80A8EE 0x90F3BB25 0x6277151C 0x9098AB63
+0x127B43E5 0xB6208EEC 0x0B895168 0xC65788D6
+0x4067BD6C 0xCD10D4B3 0xA689BA69 0x3D5C3F86
+0xB9E7694E 0x568672CB 0x4EF93A0F 0x73471E5C
+0x121CBB41 0x26775A98 0x33D278EE 0xA92202EE
+0x6EB4788C 0x8B92D176 0xB6A79E04 0x655C272A
+0xE09264D2 0xEB7575CF 0x66272B8C 0x4452A007
+0xDA070D6A 0x0912A1D6 0x7AC1D664 0x636AA549
+0x9CB5D1B6 0x47EAE2B0 0x80ECD81C 0xA29C3099
+0x034C3C2A 0xAAEC7AD9 0xD5750A6A 0xB339BB8B
+
+Key ID:                          0x43534B45		; CSKE
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x1B9BA633 0x20E1A628 0xF590848D 0x61EEE18A
+0x818E9C11 0xDB41F99E 0x60C41E9A 0x0D3F2F12
+0x25CFF022 0x40D6F206 0xBD956736 0x0CE10AF5
+0x9644161D 0x4E311CED 0xC759D55C 0xF6067A34
+0x14409A8E 0x172FB96C 0xE4DF02F1 0x9A823EA9
+0xAE011A2C 0xB0193083 0x569C00B8 0x947AEC1E
+0x1CC0A4EB 0x4EF3D0CA 0x8B618734 0xEA825D95
+0x01E741B8 0xD819A62F 0xD69C38CA 0xEC0CFD96
+0x4714DDEC 0xECD1BA71 0xDD1534FB 0x12400336
+0xB820B99C 0x5E483F3A 0xB691E5DD 0x3A65E34F
+0x8068C6D7 0x6C67F719 0x9734CB95 0x3B69268C
+0x69028406 0xC782F8D9 0xF344CC56 0xFD4696F2
+0xF3D4B293 0x50C07A75 0x419DCC90 0x7EDF9C9B
+0xDA146BF2 0x13BA7DA5 0x5B1ABC21 0xF60EC64F
+0xECC0D7ED 0xB09192C6 0xDCF4ABA5 0xC34DE401
+0x9AC70C66 0xCD16209E 0x5F2C6F9F 0xB0C05D75
+
+Key ID:                          0x43534B46		; CSKF
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+Public Key Size in bits:         2048
+Encrypt Algorithm ID:            7
+RSA Public Key Exponent:
+0x00000007 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x8BAAF09B 0x9F054D65 0xD96E8251 0xD6A54983
+0xE298F2A7 0xC3A61186 0x1569AF1C 0xCF08DF01
+0xDA22D992 0x464A7549 0xA0743F28 0x21F21FDB
+0x62883E78 0xF273C89E 0x398BA51A 0xA29A01CB
+0x1D1DF0BB 0x55D8CEEF 0xDD2F3725 0xAF498E7D
+0x1757D055 0xEF607138 0x8F576E6D 0x3986C646
+0x0D001EB0 0xD2E72201 0x24A7CEC7 0xBA1B007D
+0xF0377102 0xB913E5F2 0xA485BE45 0xB9EB9F8D
+0x316D5671 0xA8E01324 0x5B4EA32C 0x576D3154
+0xB6DB746D 0x25B9473C 0xEC73B697 0x1663B961
+0x942F4E3E 0x8F18117B 0xC37B494C 0xC1F94666
+0x822AAFFE 0x3C5456AE 0xB3691FA5 0x7299BB72
+0xE1FBD6A6 0x24C49D7F 0x36BB9FE6 0x0BE04D0F
+0x4D24A7ED 0x7F0E149F 0x015F6836 0xFCD94761
+0x70CEC997 0xCA69DC43 0x0ECF7082 0x067CB240
+0x5163D25E 0x506BB33B 0x632BCC5F 0xA56F6AE8
+
+Reserved Data:
+0x4F505448	; OPTH reserved package
+0x00000002	; Number of reserved packages
+0x494D4150	; Image Map (IMAP) Package
+0x00000020	; Size of this package
+0x00000001	; Number of image map
+0x43534B54	; ImageID: CSKT
+0x00000000	; Primary:0, Recovery:1
+0x00004000	; Flash Entry Address
+0x00000000	; Reserved
+0x00000000	; Partition # in the flash
+0x5465726D	; Terminate package
+0x00000008	; Size of this package
+
+Extended Reserved Data:
+Consumer ID:
+CID: TBRI
+PID: DDR3
+End Consumer ID:
+DDR Initialization:
+DDR_PID: DDR3
+Operations:
+DDR_INIT_ENABLE: 0x00000001
+End Operations:
+Instructions:
+WRITE: 0xC0014008 0x00404500
+WRITE: 0xC0002000 0x00010000
+WRITE: 0xC0002004 0x00000000
+WRITE: 0xC0000340 0x0f0f0fef
+WRITE: 0xC0000344 0x100000aa
+WRITE: 0xC0000310 0x200000
+WRITE: 0xC0000304 0x0
+WRITE: 0xC0000308 0x0
+WRITE: 0xC0000200 0xe0001
+WRITE: 0xC0000204 0x0
+WRITE: 0xC0000220 0x13020532
+WRITE: 0xC0000044 0x30200
+WRITE: 0xC00002c0 0x6000
+WRITE: 0xC00002c4 0x00100020
+WRITE: 0xC0000058 0x143f
+WRITE: 0xC0000048 0x1
+WRITE: 0xC0000180 0x00010200
+WRITE: 0xC0000050 0x1ff
+WRITE: 0xC000004c 0x0
+WRITE: 0xC0000054 0x480
+WRITE: 0xC0000300 0x708
+WRITE: 0xC0000380 0x1f5
+WRITE: 0xC0000384 0x3e9
+WRITE: 0xC0000388 0x9600043
+WRITE: 0xC000038c 0x200
+WRITE: 0xC0000390 0x400100
+WRITE: 0xC0000394 0x6b03cf
+WRITE: 0xC0000398 0x720200
+WRITE: 0xC000039c 0x120707
+WRITE: 0xC00003a0 0x40511
+WRITE: 0xC00003a4 0x1
+WRITE: 0xC00003a8 0xc04
+WRITE: 0xC00003ac 0x15210919
+WRITE: 0xC00003b0 0x90b0609
+WRITE: 0xC00003b4 0x4000600
+WRITE: 0xC00003b8 0x600
+WRITE: 0xC00003bc 0x2020404
+WRITE: 0xC00003c0 0x0
+WRITE: 0xC00003c4 0x0
+WRITE: 0xC00003dc 0x81239
+WRITE: 0xC00002c8 0x0
+WRITE: 0xC0000064 0x0006
+WRITE: 0xC0001004 0x10077779
+WRITE: 0xC0001008 0x1ff00770
+WRITE: 0xC000100c 0x3f03fc77
+WRITE: 0xC0001010 0x100118
+WRITE: 0xC0001028 0x0
+WRITE: 0xC0001030 0x3800000
+WRITE: 0xC0001034 0x0
+WRITE: 0xC0001040 0x400
+WRITE: 0xC00010c0 0x80000001
+WRITE: 0xC00010d0 0x0
+WRITE: 0xC00010e0 0x11ff0
+WRITE: 0xC0001090 0x0
+WRITE: 0xC0001094 0x0
+WRITE: 0xC0001098 0x0
+WRITE: 0xC000109c 0x0
+WRITE: 0xC00010a0 0x0
+WRITE: 0xC00010a4 0x0
+WRITE: 0xC00010a8 0x0
+WRITE: 0xC00010ac 0x0
+WRITE: 0xC00010b0 0x0
+WRITE: 0xC0001000 0x44041
+WRITE: 0xC0001014 0x80200
+WRITE: 0xC0001038 0x2
+WRITE: 0xC000103c 0x10
+WRITE: 0xC0001180 0x20a
+WRITE: 0xC0001184 0x20a
+WRITE: 0xC0001188 0x20a
+WRITE: 0xC000118c 0x20a
+WRITE: 0xC0001190 0x20a
+WRITE: 0xC0001194 0x20a
+WRITE: 0xC0001198 0x20a
+WRITE: 0xC000119c 0x20a
+WRITE: 0xC00011a0 0x20a
+WRITE: 0xC0001050 0x8080000
+WRITE: 0xC0001054 0x8080000
+WRITE: 0xC0001058 0x8080000
+WRITE: 0xC000105c 0x8080000
+WRITE: 0xC0001060 0x8080000
+WRITE: 0xC0001064 0x8080000
+WRITE: 0xC0001068 0x8080000
+WRITE: 0xC000106c 0x8080000
+WRITE: 0xC0001070 0x8080000
+WRITE: 0xC0001020 0x20000000
+WRITE: 0xC0001020 0x40000000
+WRITE: 0xC0001020 0x80000000
+WRITE: 0xC0000020 0x11000001
+WAIT_FOR_BIT_SET: 0xC0000008 0x00000001 0x00001000
+WRITE: 0x60030000 0x01234567
+End Instructions:
+End DDR Initialization:
+End Extended Reserved Data:
+
+DSA Algorithm ID:                7			; Signed with KAK
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+RSA Public Exponent:
+0x00000005 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0x77F56E6F 0xF5DB0AD4 0xEB994D35 0xA863FE85
+0x747D27ED 0x9EE303BA 0xFB55D06C 0x8DBAEF04
+0x9AC8F88F 0x8867FA62 0x8DFA5127 0x286D8EA5
+0x27FF8E37 0x70352983 0x8C91B233 0xF405C5D9
+0xA6A5C82C 0x837A1F87 0xBA48B0C1 0x93F53C48
+0x28D7FC8A 0xF9C7ADA9 0x670C502A 0x1FC12DA2
+0x71731332 0x0EB4C539 0x302C6048 0xF509D000
+0x76652F8E 0x934F1993 0x1FA1B34E 0xD16CB85A
+0x158676A5 0xE7C149B0 0xC205E905 0x7E93ABD2
+0x8EED7C51 0xCBB4B5DB 0x46B79B5D 0xFE93923A
+0x19880137 0xA3B4AD2E 0x0F1052D8 0xA603C887
+0x574BB1E8 0xE9BCA9D7 0xB895700B 0x56D44368
+0x8A98798E 0xFB40038D 0xB6486D0F 0x5328EC7D
+0x888616A8 0xBCD39C2C 0x90274217 0x0D902683
+0x2E195296 0x661D62BC 0x278042A2 0x30ECAD89
+0x6A2541EF 0xA94696C9 0x21482342 0x8B5E9CE9
+RSA Private Key:
+0xD4AFB845 0x882291D2 0x0AEE53EB 0xF1A487FB
+0xD4C8B4A3 0x62FF05F1 0x316CB1CE 0x88C4E130
+0xA0DFD69F 0xBBB22559 0xB8EE1753 0xB61D13D3
+0xB2A0B791 0x83EFABDF 0x28E75756 0x6DFBF10E
+0xC4022294 0x403D1E65 0x977AE60F 0xF1286EF7
+0x77881617 0xCD439140 0x805117C3 0x8CA829F3
+0x29BBC428 0x170A0A39 0xD10CD562 0x0B156871
+0xDB551DE2 0xFFE0D58E 0xE7FEDD0E 0x77037604
+0x044E17BA 0x94C041F0 0x26CDFB67 0x4C83EF2A
+0x82FC4C10 0xF58A8AC5 0xA7BE5245 0xCC83EA0B
+0x384E66A4 0xED8A8909 0x030343C4 0x5467281B
+0x117589FB 0x2EBF552B 0xF1B77CCF 0x115DA714
+0x821E7EB6 0xCBD99A4F 0x2474E29C 0x43D4FC19
+0x4E8137BB 0x8C2A526F 0x1CD4A6D1 0xCF833AE7
+0x3C6B76EA 0xE13913BF 0xD4B34086 0x09C8EF81
+0xAED44063 0xBB74848E 0x6D0E6D73 0x1BDFB8FB
diff --git a/tools/wtp/u-boot-timN.txt b/tools/wtp/u-boot-timN.txt
new file mode 100644
index 0000000..89d34ea
--- /dev/null
+++ b/tools/wtp/u-boot-timN.txt
@@ -0,0 +1,126 @@
+Version:                        0x00030600
+Trusted:                        0x00000001
+Issue Date:                     0x12162015		; today's date
+OEM UniqueID:                   0x4D52564C		; MRVL
+Stepping:                       0x00000000
+Processor Type:                 <undefined>		; ARMADALP
+Boot Flash Signature:           0x5350490A		; Select SPI'0A
+Number of Images:               4			; TIMN + OBMI +WTMI + OSLO
+Number of Keys:                 0			; DSIG key is not counted
+Size of Reserved in bytes:      44
+
+Image ID:                       0x54494D4E		; TIMN
+Next Image ID:                  0x4F424d49		; Next Image - OBMI
+Flash Entry Address:            0x00004000		; TIMN flash addr
+Load Address:                   0x20004100		; TIMN ISRAM addr
+Image Size To Hash in bytes:    0xFFFFFFFF		; Hash complete image file
+Hash Algorithm ID:              32			; SHA-256
+Partition Number:               0x00000000
+Encrypt Algorithm ID:           0x00000000
+Encrypt Start Offset:           0x00000000
+Encrypt Size:                   0x00000000
+Image Filename: TIMN_UBOOT_TRUSTED.bin
+
+Image ID:                       0x4F424d49		; OBMI - 64K code + 16K data
+Next Image ID:                  0x57544d49		; WTMI next
+Flash Entry Address:            0x00008000		; OBMI flash addr
+Load Address:                   0x1FFF0000		; OBMI RAM addr
+Image Size To Hash in bytes:    0xFFFFFFFF		; Hash complete image file
+Hash Algorithm ID:              32			; SHA-256
+Partition Number:               0x00000000
+Encrypt Algorithm ID:           0x00000000
+Encrypt Start Offset:           0x00000000
+Encrypt Size:                   0x00000000
+Image Filename: spl/u-boot-spl.bin
+
+Image ID:                       0x57544d49		; WTMI
+Next Image ID:                  0x4F534C4F		; OSLO next
+Flash Entry Address:            0x00028000		; WTMI flash addr
+Load Address:                   0x200040F0		; WTMI RAM addr
+Image Size To Hash in bytes:    0xFFFFFFFF		; Hash complete image file
+Hash Algorithm ID:              32			; SHA-256
+Partition Number:               0x00000000
+Encrypt Algorithm ID:           0x00000000
+Encrypt Start Offset:           0x00000000
+Encrypt Size:                   0x00000000
+Image Filename: tools/wtp/wtmi.dummy
+
+Image ID:                       0x4F534C4F		; OSLO
+Next Image ID:                  0xFFFFFFFF		; NONE
+Flash Entry Address:            0x0002A000		; OSLO flash addr
+Load Address:                   0x60000000		; OSLO RAM addr
+Image Size To Hash in bytes:    0xFFFFFFFF		; Hash complete image file
+Hash Algorithm ID:              32			; SHA-256
+Partition Number:               0x00000000
+Encrypt Algorithm ID:           0x00000000
+Encrypt Start Offset:           0x00000000
+Encrypt Size:                   0x00000000
+Image Filename: u-boot.bin
+
+Reserved Data:
+0x4F505448	;OPTH
+0x00000002	; Number of Reserved Packages
+0x43525632	;CRV2
+0x0000001C	; Size of CRV2 package in bytes
+0x00000002	; Number of cores to release
+0x00000000	; Core_ID: 0 - AP0, 1 - AP1, 2 - CM3
+0xF0000000	; PC address for Core_ID above
+0x00000002	; Core_ID: 0 - AP0, 1 - AP1, 2 - CM3
+0x200040F0	; PC address for Core_ID above
+0x5465726D	;Term
+0x00000008	; Size of this package in bytes
+
+DSA Algorithm ID:                7			; Signed with CSK3
+Hash Algorithm ID:               32
+Key Size in bits:                2048
+RSA Public Exponent:
+0x00000003 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+0x00000000 0x00000000 0x00000000 0x00000000
+RSA System Modulus:
+0xCC14894B 0x1D54EFAB 0x2C006ECC 0xECBA39B5
+0x386B9504 0xE5127FA9 0xBBAF3A06 0xB351C24B
+0x90A1B218 0xDB877D1E 0xFAAECD8C 0x2ABDC223
+0x88AB687E 0x3BB63A7A 0xE2CECAA9 0x9F51C0D8
+0xF55351ED 0x68C4C09E 0xF4FED718 0x111E24B9
+0x8D938BEB 0x8ECF27F9 0xBBE7D8D5 0x1AC4F691
+0xD54FBDB1 0x18A6D6BE 0x9619F30B 0x96461657
+0xDF2961EC 0x3AE58790 0x88ECC5BD 0x181A4B15
+0xED03EB8E 0xB365ED3E 0x7C95DCEA 0xE13FEF7A
+0x046073C2 0xDF910B26 0x3F629BFD 0xDA8AD3F3
+0xE3FBCFD8 0x49338817 0x8BBF17BD 0x02AAEAFA
+0x4678A53E 0xB7D51770 0xBF853DCA 0x75487D2D
+0xD57ECF69 0x9E444CF0 0x0B7B1D0D 0x38C1DEE9
+0xEDC2BDB1 0xDF03CEA1 0xDDD31093 0x0317A73D
+0x02EB8F74 0x5519D9F9 0xA3501FDE 0xC4BA749D
+0xA82D50A6 0x0E42C2C6 0x0E56FC52 0xA533A925
+RSA Private Key:
+0xC943324B 0xCE30A122 0x4A1B377D 0xA80E4F16
+0xFFB441C8 0x196E7696 0x32ADDC6C 0xF2A97A06
+0x754B38D7 0x0339DA63 0xA10C8866 0x6A96C338
+0x883DA2BA 0x12D95E0A 0xA8C66443 0xF31C28AC
+0x02217A27 0x44A5611A 0xD47EEDC5 0xBF607912
+0x45DCE74B 0x369C0D05 0x125C8DBE 0x520AA3AE
+0xA2D82F88 0x96B5F30E 0x14812111 0x5F349720
+0xC636D203 0xE68D9D63 0x1E7FCC03 0x5086A930
+0x9E029D08 0x779948D4 0xA863E89C 0x40D54A51
+0x02EAF7D7 0x950B5CC4 0xD4EC67FE 0x3C5C8D4C
+0x97FD353B 0x86225ABA 0x5D2A0FD3 0xAC71F1FC
+0xD9A5C37E 0x2538BA4A 0xD5037E87 0x4E305373
+0x8E548A46 0x142D88A0 0x5CFCBE09 0x25D6949B
+0x492C7E76 0xEA0289C1 0x3E8CB5B7 0x57651A29
+0xAC9D0A4D 0xE3669150 0x6CE0153E 0xD87C4DBE
+0xC5738B19 0xB42C81D9 0x5EE4A836 0x6E2270C3
-- 
1.9.1

